// ngx-spinners/bundles/ngx-spinners.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ngx-spinners', ['exports', '@angular/core', '@angular/common'], factory) :
    (global = global || self, factory(global['ngx-spinners'] = {}, global.ng.core, global.ng.common));
}(this, function (exports, core, common) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SyncLoaderComponent = /** @class */ (function () {
        function SyncLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @param {?} index
         * @return {?}
         */
        SyncLoaderComponent.prototype.createBallStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'display': 'inline-block',
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': '100%',
                'animation': "loading-sync 0.6s " + index * 0.07 + "s infinite ease-in-out",
                'animation-fill-mode': 'both'
            };
        };
        SyncLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-sync-loader',
                        template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createBallStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(2)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(3)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-sync{33%{transform:translateY(10px)}66%{transform:translateY(-10px)}100%{transform:translateY(0)}}@keyframes loading-sync{33%{transform:translateY(10px)}66%{transform:translateY(-10px)}100%{transform:translateY(0)}}"]
                    }] }
        ];
        /** @nocollapse */
        SyncLoaderComponent.ctorParameters = function () { return []; };
        SyncLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return SyncLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        SyncLoaderComponent.prototype.color;
        /** @type {?} */
        SyncLoaderComponent.prototype.size;
        /** @type {?} */
        SyncLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        SyncLoaderComponent.prototype.margin;
        /** @type {?} */
        SyncLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SquareLoaderComponent = /** @class */ (function () {
        function SquareLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @return {?}
         */
        SquareLoaderComponent.prototype.createBoxStyle = /**
         * @return {?}
         */
        function () {
            return {
                'display': 'inline-block',
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'animation': 'loading-square 3s 0s infinite cubic-bezier(.09,.57,.49,.9)',
                'animation-fill-mode': 'both'
            };
        };
        SquareLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-square-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createBoxStyle()\"></div>\r\n",
                        styles: ["@-webkit-keyframes loading-square{25%{transform:rotateX(180deg) rotateY(0)}50%{transform:rotateX(180deg) rotateY(180deg)}75%{transform:rotateX(0) rotateY(180deg)}100%{transform:rotateX(0) rotateY(0)}}@keyframes loading-square{25%{transform:rotateX(180deg) rotateY(0)}50%{transform:rotateX(180deg) rotateY(180deg)}75%{transform:rotateX(0) rotateY(180deg)}100%{transform:rotateX(0) rotateY(0)}}"]
                    }] }
        ];
        /** @nocollapse */
        SquareLoaderComponent.ctorParameters = function () { return []; };
        SquareLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return SquareLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        SquareLoaderComponent.prototype.color;
        /** @type {?} */
        SquareLoaderComponent.prototype.size;
        /** @type {?} */
        SquareLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        SquareLoaderComponent.prototype.margin;
        /** @type {?} */
        SquareLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var SkewLoaderComponent = /** @class */ (function () {
        function SkewLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        SkewLoaderComponent.prototype.createSkewStyle = /**
         * @return {?}
         */
        function () {
            return {
                'display': 'inline-block',
                'width': '0',
                'height': '0',
                'border-left': "" + this.size + this.sizeUnit + " solid transparent",
                'border-right': "" + this.size + this.sizeUnit + " solid transparent",
                'border-bottom': "" + this.size + this.sizeUnit + " solid " + this.color,
                'animation': 'loading-skew 3s 0s infinite cubic-bezier(.09,.57,.49,.9)',
                'animation-fill-mode': 'both'
            };
        };
        SkewLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-skew-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createSkewStyle()\"></div>\r\n",
                        styles: ["@-webkit-keyframes loading-skew{25%{transform:perspective(100px) rotateX(180deg) rotateY(0)}50%{transform:perspective(100px) rotateX(180deg) rotateY(180deg)}75%{transform:perspective(100px) rotateX(0) rotateY(180deg)}100%{transform:perspective(100px) rotateX(0) rotateY(0)}}@keyframes loading-skew{25%{transform:perspective(100px) rotateX(180deg) rotateY(0)}50%{transform:perspective(100px) rotateX(180deg) rotateY(180deg)}75%{transform:perspective(100px) rotateX(0) rotateY(180deg)}100%{transform:perspective(100px) rotateX(0) rotateY(0)}}"]
                    }] }
        ];
        /** @nocollapse */
        SkewLoaderComponent.ctorParameters = function () { return []; };
        SkewLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return SkewLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        SkewLoaderComponent.prototype.color;
        /** @type {?} */
        SkewLoaderComponent.prototype.size;
        /** @type {?} */
        SkewLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ScaleLoaderComponent = /** @class */ (function () {
        function ScaleLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.width = 4;
            this.widthUnit = 'px';
            this.height = 35;
            this.heightUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
            this.radius = 2;
            this.radiusUnit = 'px';
        }
        /**
         * @param {?} index
         * @return {?}
         */
        ScaleLoaderComponent.prototype.createScaleStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'display': 'inline-block',
                'background-color': "" + this.color,
                'width': "" + ("" + this.width + this.widthUnit),
                'height': "" + ("" + this.height + this.heightUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': "" + ("" + this.radius + this.radiusUnit),
                'animation': "loading-scale 1s " + index * 0.1 + "s infinite cubic-bezier(.2,.68,.18,1.08)",
                'animation-fill-mode': 'both'
            };
        };
        ScaleLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-scale-loader',
                        template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createScaleStyle(1)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(2)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(3)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(4)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(5)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-scale{0%,100%{transform:scaleY(1)}50%{transform:scaleY(.4)}}@keyframes loading-scale{0%,100%{transform:scaleY(1)}50%{transform:scaleY(.4)}}"]
                    }] }
        ];
        /** @nocollapse */
        ScaleLoaderComponent.ctorParameters = function () { return []; };
        ScaleLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            width: [{ type: core.Input }],
            widthUnit: [{ type: core.Input }],
            height: [{ type: core.Input }],
            heightUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }],
            radius: [{ type: core.Input }],
            radiusUnit: [{ type: core.Input }]
        };
        return ScaleLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        ScaleLoaderComponent.prototype.color;
        /** @type {?} */
        ScaleLoaderComponent.prototype.width;
        /** @type {?} */
        ScaleLoaderComponent.prototype.widthUnit;
        /** @type {?} */
        ScaleLoaderComponent.prototype.height;
        /** @type {?} */
        ScaleLoaderComponent.prototype.heightUnit;
        /** @type {?} */
        ScaleLoaderComponent.prototype.margin;
        /** @type {?} */
        ScaleLoaderComponent.prototype.marginUnit;
        /** @type {?} */
        ScaleLoaderComponent.prototype.radius;
        /** @type {?} */
        ScaleLoaderComponent.prototype.radiusUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var RotateLoaderComponent = /** @class */ (function () {
        function RotateLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @return {?}
         */
        RotateLoaderComponent.prototype.createRotateWrapperStyle = /**
         * @return {?}
         */
        function () {
            return __assign({ 'display': 'inline-block' }, this.createBallStyle(), { 'position': 'relative', 'animation-fill-mode': 'both', 'animation': 'loading-rotate 1s 0s infinite cubic-bezier(.7,-.13,.22,.86)' });
        };
        /**
         * @return {?}
         */
        RotateLoaderComponent.prototype.createRotateLongStyle = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createBallStyle(), this.createPositionStyle(1));
        };
        /**
         * @return {?}
         */
        RotateLoaderComponent.prototype.createRotateShortStyle = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createBallStyle(), this.createPositionStyle(2));
        };
        /**
         * @private
         * @return {?}
         */
        RotateLoaderComponent.prototype.createBallStyle = /**
         * @private
         * @return {?}
         */
        function () {
            return {
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': '100%'
            };
        };
        /**
         * @private
         * @param {?} index
         * @return {?}
         */
        RotateLoaderComponent.prototype.createPositionStyle = /**
         * @private
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'opacity': '0.8',
                'position': 'absolute',
                'top': '0',
                'left': (index % 2 ? -28 : 25) + "px"
            };
        };
        RotateLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-rotate-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createRotateWrapperStyle()\">\r\n  <div [ngStyle]=\"createRotateLongStyle()\"></div>\r\n  <div [ngStyle]=\"createRotateShortStyle()\"></div>\r\n</div>\r\n",
                        encapsulation: core.ViewEncapsulation.ShadowDom,
                        styles: ["@-webkit-keyframes loading-rotate{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}@keyframes loading-rotate{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        RotateLoaderComponent.ctorParameters = function () { return []; };
        RotateLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return RotateLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        RotateLoaderComponent.prototype.color;
        /** @type {?} */
        RotateLoaderComponent.prototype.size;
        /** @type {?} */
        RotateLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        RotateLoaderComponent.prototype.margin;
        /** @type {?} */
        RotateLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var RiseLoaderComponent = /** @class */ (function () {
        function RiseLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @param {?} index
         * @return {?}
         */
        RiseLoaderComponent.prototype.createRiseStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'display': 'inline-block',
                'border-radius': '100%',
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'animation': (index % 2 === 0 ? 'loading-rise-even' : 'loading-rise-odd') + " 1s 0s infinite cubic-bezier(.15,.46,.9,.6)",
                'animation-fill-mode': 'both'
            };
        };
        RiseLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-rise-loader',
                        template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createRiseStyle(1)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(2)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(3)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(4)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(5)\"></div>\r\n</div>\r\n",
                        encapsulation: core.ViewEncapsulation.ShadowDom,
                        styles: [":host{--riseAmount:30px}@-webkit-keyframes loading-rise-even{0%{transform:scale(1.1)}50%{transform:scale(.4)}75%{transform:translateY(var(--riseAmount))}100%{transform:translateY(0) scale(1)}}@keyframes loading-rise-even{0%{transform:scale(1.1)}50%{transform:scale(.4)}75%{transform:translateY(var(--riseAmount))}100%{transform:translateY(0) scale(1)}}@-webkit-keyframes loading-rise-odd{0%{transform:scale(.4)}50%{transform:scale(1.1)}75%{transform:translateY(calc(-1 * var(--riseAmount)))}100%{transform:translateY(0) scale(.75)}}@keyframes loading-rise-odd{0%{transform:scale(.4)}50%{transform:scale(1.1)}75%{transform:translateY(calc(-1 * var(--riseAmount)))}100%{transform:translateY(0) scale(.75)}}"]
                    }] }
        ];
        /** @nocollapse */
        RiseLoaderComponent.ctorParameters = function () { return []; };
        RiseLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return RiseLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        RiseLoaderComponent.prototype.color;
        /** @type {?} */
        RiseLoaderComponent.prototype.size;
        /** @type {?} */
        RiseLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        RiseLoaderComponent.prototype.margin;
        /** @type {?} */
        RiseLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var RingLoaderComponent = /** @class */ (function () {
        function RingLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 60;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        RingLoaderComponent.prototype.createRingWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'position': 'relative'
            };
        };
        /**
         * @param {?} index
         * @return {?}
         */
        RingLoaderComponent.prototype.createRingStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'position': 'absolute',
                'top': '0',
                'left': '0',
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'border': "" + this.size / 10 + this.sizeUnit + " solid " + this.color,
                'opacity': '0.4',
                'border-radius': '100%',
                'animation-fill-mode': 'forwards',
                'perspective': '800px',
                'animation': (index === 1 ? 'loading-ring-right' : 'loading-ring-left') + " 2s 0s infinite linear"
            };
        };
        RingLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-ring-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createRingWrapperStyle()\">\r\n  <div [ngStyle]=\"createRingStyle(1)\"></div>\r\n  <div [ngStyle]=\"createRingStyle(2)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-ring-right{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(180deg) rotateY(360deg) rotateZ(360deg)}}@keyframes loading-ring-right{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(180deg) rotateY(360deg) rotateZ(360deg)}}@-webkit-keyframes loading-ring-left{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(360deg) rotateY(180deg) rotateZ(360deg)}}@keyframes loading-ring-left{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(360deg) rotateY(180deg) rotateZ(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        RingLoaderComponent.ctorParameters = function () { return []; };
        RingLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return RingLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        RingLoaderComponent.prototype.color;
        /** @type {?} */
        RingLoaderComponent.prototype.size;
        /** @type {?} */
        RingLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PulseLoaderComponent = /** @class */ (function () {
        function PulseLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @param {?} index
         * @return {?}
         */
        PulseLoaderComponent.prototype.createPulseStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'display': 'inline-block',
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': '100%',
                'animation': "loading-pulse 0.75s " + index * 0.12 + "s infinite cubic-bezier(.2,.68,.18,1.08)",
                'animation-fill-mode': 'both'
            };
        };
        PulseLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-pulse-loader',
                        template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createPulseStyle(1)\"></div>\r\n  <div [ngStyle]=\"createPulseStyle(2)\"></div>\r\n  <div [ngStyle]=\"createPulseStyle(3)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-pulse{0%,80%{transform:scale(1);opacity:1}45%{transform:scale(.1);opacity:.7}}@keyframes loading-pulse{0%,80%{transform:scale(1);opacity:1}45%{transform:scale(.1);opacity:.7}}"]
                    }] }
        ];
        /** @nocollapse */
        PulseLoaderComponent.ctorParameters = function () { return []; };
        PulseLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return PulseLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        PulseLoaderComponent.prototype.color;
        /** @type {?} */
        PulseLoaderComponent.prototype.size;
        /** @type {?} */
        PulseLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        PulseLoaderComponent.prototype.margin;
        /** @type {?} */
        PulseLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PropagateLoaderComponent = /** @class */ (function () {
        function PropagateLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        PropagateLoaderComponent.prototype.createPropagateWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative'
            };
        };
        /**
         * @param {?} index
         * @return {?}
         */
        PropagateLoaderComponent.prototype.createPropagateStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'position': 'absolute',
                'font-size': "" + ("" + this.size / 3 + this.sizeUnit),
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'background': "" + this.color,
                'border-radius': '50%',
                'animation': 'loading-propagate-' + index + " 1.5s  infinite",
                'animation-fill-mode': 'forwards'
            };
        };
        PropagateLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-propagate-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createPropagateWrapperStyle()\">\r\n  <div [ngStyle]=\"createPropagateStyle(0)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(1)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(2)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(3)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(4)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(5)\"></div>\r\n</div>\r\n",
                        styles: [":host{--oneRem:1rem;--threeRem:3rem;--fiveRem:5rem}@-webkit-keyframes loading-propagate-0{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}75%{transform:translateX(calc(-1 * var(--fiveRem))) scale(.5)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-0{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}75%{transform:translateX(calc(-1 * var(--fiveRem))) scale(.5)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-1{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%,75%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-1{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%,75%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-2{25%,75%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-2{25%,75%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-3{25%,75%{transform:translateX(var(--oneRem)) scale(.75)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-3{25%,75%{transform:translateX(var(--oneRem)) scale(.75)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-4{25%{transform:translateX(var(--oneRem)) scale(.75)}50%,75%{transform:translateX(var(--threeRem)) scale(.6)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-4{25%{transform:translateX(var(--oneRem)) scale(.75)}50%,75%{transform:translateX(var(--threeRem)) scale(.6)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-5{25%{transform:translateX(var(--oneRem)) scale(.75)}50%{transform:translateX(var(--threeRem)) scale(.6)}75%{transform:translateX(var(--fiveRem)) scale(.5)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-5{25%{transform:translateX(var(--oneRem)) scale(.75)}50%{transform:translateX(var(--threeRem)) scale(.6)}75%{transform:translateX(var(--fiveRem)) scale(.5)}95%{transform:translateX(0) scale(1)}}"]
                    }] }
        ];
        /** @nocollapse */
        PropagateLoaderComponent.ctorParameters = function () { return []; };
        PropagateLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return PropagateLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        PropagateLoaderComponent.prototype.color;
        /** @type {?} */
        PropagateLoaderComponent.prototype.size;
        /** @type {?} */
        PropagateLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var PacmanLoaderComponent = /** @class */ (function () {
        function PacmanLoaderComponent(elementRef) {
            this.elementRef = elementRef;
            this.color = 'rgb(54, 215, 183)';
            this.size = 25;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @param {?} changes
         * @return {?}
         */
        PacmanLoaderComponent.prototype.ngOnChanges = /**
         * @param {?} changes
         * @return {?}
         */
        function (changes) {
            var size = changes.size, sizeUnit = changes.sizeUnit;
            if (size && !isNaN(size.currentValue) && sizeUnit && typeof sizeUnit.currentValue === 'string') {
                this.elementRef.nativeElement.style.setProperty('--tx', "" + ("" + -4 * size.currentValue + sizeUnit.currentValue));
                this.elementRef.nativeElement.style.setProperty('--ty', "" + ("" + (-1 * size.currentValue) / 4 + sizeUnit.currentValue));
            }
        };
        /**
         * @return {?}
         */
        PacmanLoaderComponent.prototype.createPacmanWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'font-size': '0',
                'height': "" + ("" + this.size + this.sizeUnit),
                'width': "" + ("" + this.size + this.sizeUnit)
            };
        };
        /**
         * @return {?}
         */
        PacmanLoaderComponent.prototype.createPacStyle = /**
         * @return {?}
         */
        function () {
            return this.createPacManStyle(0);
        };
        /**
         * @return {?}
         */
        PacmanLoaderComponent.prototype.createManStyle = /**
         * @return {?}
         */
        function () {
            return this.createPacManStyle(1);
        };
        /**
         * @param {?} index
         * @return {?}
         */
        PacmanLoaderComponent.prototype.createBallStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'width': "" + ("" + this.size / 3 + this.sizeUnit),
                'height': "" + ("" + this.size / 3 + this.sizeUnit),
                'background-color': "" + this.color,
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': '100%',
                'transform': "translate(0, " + ("" + -this.size / 4 + this.sizeUnit) + ")",
                'position': 'absolute',
                'top': this.size + "px",
                'left': "" + ("" + this.size * 4 + this.sizeUnit),
                'animation': "loading-ball 1s " + index * 0.25 + "s infinite linear",
                'animation-fill-mode': 'both'
            };
        };
        /**
         * @private
         * @param {?} index
         * @return {?}
         */
        PacmanLoaderComponent.prototype.createPacManStyle = /**
         * @private
         * @param {?} index
         * @return {?}
         */
        function (index) {
            /** @type {?} */
            var s1 = this.s1();
            /** @type {?} */
            var s2 = this.s2();
            return {
                'width': '0',
                'height': '0',
                'border-right': "" + s1,
                'border-top': "" + (index === 0 ? s1 : s2),
                'border-left': "" + s2,
                'border-bottom': "" + (index === 0 ? s2 : s1),
                'border-radius': "" + ("" + this.size + this.sizeUnit),
                'position': 'absolute',
                'animation': 'loading-pacman-' + index + " 0.8s infinite ease-in-out",
                'animation-fill-mode': 'both'
            };
        };
        /**
         * @private
         * @return {?}
         */
        PacmanLoaderComponent.prototype.s1 = /**
         * @private
         * @return {?}
         */
        function () {
            return "" + this.size + this.sizeUnit + " solid transparent";
        };
        /**
         * @private
         * @return {?}
         */
        PacmanLoaderComponent.prototype.s2 = /**
         * @private
         * @return {?}
         */
        function () {
            return "" + this.size + this.sizeUnit + " solid " + this.color;
        };
        PacmanLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-pacman-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createPacmanWrapperStyle()\">\r\n  <div [ngStyle]=\"createPacStyle()\"></div>\r\n  <div [ngStyle]=\"createManStyle()\"></div>\r\n  <div [ngStyle]=\"createBallStyle(2)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(3)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(4)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(5)\"></div>\r\n</div>\r\n",
                        styles: [":host{--tx:calc(-4 * 25px);--ty:calc((-1 * 25px) / 4)}@-webkit-keyframes loading-ball{75%{opacity:.7}100%{transform:translate(var(--tx),var(--ty))}}@keyframes loading-ball{75%{opacity:.7}100%{transform:translate(var(--tx),var(--ty))}}@-webkit-keyframes loading-pacman-0{0%{transform:rotate(0)}50%{transform:rotate(-44deg)}}@keyframes loading-pacman-0{0%{transform:rotate(0)}50%{transform:rotate(-44deg)}}@-webkit-keyframes loading-pacman-1{0%{transform:rotate(0)}50%{transform:rotate(44deg)}}@keyframes loading-pacman-1{0%{transform:rotate(0)}50%{transform:rotate(44deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        PacmanLoaderComponent.ctorParameters = function () { return [
            { type: core.ElementRef }
        ]; };
        PacmanLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return PacmanLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        PacmanLoaderComponent.prototype.color;
        /** @type {?} */
        PacmanLoaderComponent.prototype.size;
        /** @type {?} */
        PacmanLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        PacmanLoaderComponent.prototype.margin;
        /** @type {?} */
        PacmanLoaderComponent.prototype.marginUnit;
        /**
         * @type {?}
         * @private
         */
        PacmanLoaderComponent.prototype.elementRef;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var MoonLoaderComponent = /** @class */ (function () {
        function MoonLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 60;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        MoonLoaderComponent.prototype.createMoonWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': "" + ("" + (this.size + ((this.size / 7) * 2)) + this.sizeUnit),
                'height': "" + ("" + (this.size + ((this.size / 7) * 2)) + this.sizeUnit),
                'animation': 'loading-moon 0.6s 0s infinite linear',
                'animation-fill-mode': 'forwards'
            };
        };
        /**
         * @return {?}
         */
        MoonLoaderComponent.prototype.createMoonBallStyle = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createBallStyle(this.size / 7), { 'background-color': "" + this.color, 'opacity': '0.8', 'position': 'absolute', 'top': "" + ("" + ((this.size / 2) - (this.size / 14)) + this.sizeUnit), 'animation': 'loading-moon 0.6s 0s infinite linear', 'animation-fill-mode': 'forwards' });
        };
        /**
         * @return {?}
         */
        MoonLoaderComponent.prototype.createMoonCircleStyle = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createBallStyle(this.size), { 'border': this.size / 7 + "px solid " + this.color, 'opacity': '0.1' });
        };
        /**
         * @private
         * @param {?} size
         * @return {?}
         */
        MoonLoaderComponent.prototype.createBallStyle = /**
         * @private
         * @param {?} size
         * @return {?}
         */
        function (size) {
            return {
                'width': "" + ("" + size + this.sizeUnit),
                'height': "" + ("" + size + this.sizeUnit),
                'border-radius': '100%',
                'box-sizing': 'content-box'
            };
        };
        MoonLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-moon-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createMoonWrapperStyle()\">\r\n  <div [ngStyle]=\"createMoonBallStyle()\"></div>\r\n  <div [ngStyle]=\"createMoonCircleStyle()\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-moon{100%{transform:rotate(360deg)}}@keyframes loading-moon{100%{transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        MoonLoaderComponent.ctorParameters = function () { return []; };
        MoonLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return MoonLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        MoonLoaderComponent.prototype.color;
        /** @type {?} */
        MoonLoaderComponent.prototype.size;
        /** @type {?} */
        MoonLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var calculateRgba = (/**
     * @param {?} color
     * @param {?} opacity
     * @return {?}
     */
    function (color, opacity) {
        if (color[0] === '#') {
            color = color.slice(1);
        }
        if (color.length === 3) {
            /** @type {?} */
            var res_1 = '';
            color.split('').forEach((/**
             * @param {?} c
             * @return {?}
             */
            function (c) {
                res_1 += c;
                res_1 += c;
            }));
            color = res_1;
        }
        /** @type {?} */
        var rgbValues = color.match(/.{2}/g).map((/**
         * @param {?} hex
         * @return {?}
         */
        function (hex) { return parseInt(hex, 16); })).join(', ');
        return "rgba(" + rgbValues + ", " + opacity + ")";
    });

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var HashLoaderComponent = /** @class */ (function () {
        function HashLoaderComponent(elementRef) {
            this.elementRef = elementRef;
            this.color = '#36d7b7';
            this.size = 50;
            this.sizeUnit = 'px';
            this.rgba = (/**
             * @param {?} color
             * @return {?}
             */
            function (color) { return calculateRgba(color, 0.75); });
            this.thickness = (/**
             * @param {?} size
             * @return {?}
             */
            function (size) { return size / 5; });
        }
        /**
         * @return {?}
         */
        HashLoaderComponent.prototype.createHashWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'transform': 'rotate(165deg)'
            };
        };
        /**
         * @param {?} index
         * @param {?} color
         * @param {?} size
         * @param {?} sizeUnit
         * @return {?}
         */
        HashLoaderComponent.prototype.createHashStyle = /**
         * @param {?} index
         * @param {?} color
         * @param {?} size
         * @param {?} sizeUnit
         * @return {?}
         */
        function (index, color, size, sizeUnit) {
            this.setStyleVariables(color, size, sizeUnit);
            return {
                'display': 'block',
                'position': 'absolute',
                'content': '',
                'top': '50%',
                'left': '50%',
                'width': "" + ("" + this.size / 5 + this.sizeUnit),
                'height': "" + ("" + this.size / 5 + this.sizeUnit),
                'border-radius': "" + ("" + this.size / 10 + this.sizeUnit),
                'transform': 'translate(-50%, -50%)',
                'animation-fill-mode': 'none',
                'animation': (index === 1 ? 'loading-hash-before' : 'loading-hash-after') + " 2s infinite"
            };
        };
        /**
         * @private
         * @param {?} size
         * @return {?}
         */
        HashLoaderComponent.prototype.lat = /**
         * @private
         * @param {?} size
         * @return {?}
         */
        function (size) {
            return (this.size - this.thickness(size)) / 2;
        };
        /**
         * @private
         * @param {?} size
         * @return {?}
         */
        HashLoaderComponent.prototype.offset = /**
         * @private
         * @param {?} size
         * @return {?}
         */
        function (size) {
            return this.lat(size) - this.thickness(size);
        };
        /**
         * @private
         * @param {?} color
         * @param {?} size
         * @param {?} sizeUnit
         * @return {?}
         */
        HashLoaderComponent.prototype.setStyleVariables = /**
         * @private
         * @param {?} color
         * @param {?} size
         * @param {?} sizeUnit
         * @return {?}
         */
        function (color, size, sizeUnit) {
            /** @type {?} */
            var rgba = this.rgba(color);
            /** @type {?} */
            var lat = this.lat(size);
            /** @type {?} */
            var thickness = this.thickness(size);
            /** @type {?} */
            var offset = this.offset(size);
            this.elementRef.nativeElement.style.setProperty('--color', rgba);
            this.elementRef.nativeElement.style.setProperty('--size', "" + ("" + size + sizeUnit));
            this.elementRef.nativeElement.style.setProperty('--lat', "" + (lat + "px"));
            this.elementRef.nativeElement.style.setProperty('--thickness', "" + (thickness + "px"));
            this.elementRef.nativeElement.style.setProperty('--offset', "" + (offset + "px"));
        };
        HashLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-hash-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createHashWrapperStyle()\">\r\n      <div [ngStyle]=\"createHashStyle(1,color, size, sizeUnit)\"></div>\r\n      <div [ngStyle]=\"createHashStyle(2, color, size, sizeUnit)\"></div>\r\n</div>\r\n",
                        styles: [":host{--color:rgba(54, 215, 183, 0.75);--thickness:10px;--size:50px;--lat:20px;--offset:10px}@-webkit-keyframes loading-hash-before{0%{width:var(--thickness);box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}35%{width:var(--size);box-shadow:0 calc(-1 * var(--offset)) var(--color),0 var(--offset) var(--color)}70%{width:var(--thickness);box-shadow:calc(-1 * var(--lat)) calc(-1 * var(--offset)) var(--color),var(--lat) var(--offset) var(--color)}100%{box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}}@keyframes loading-hash-before{0%{width:var(--thickness);box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}35%{width:var(--size);box-shadow:0 calc(-1 * var(--offset)) var(--color),0 var(--offset) var(--color)}70%{width:var(--thickness);box-shadow:calc(-1 * var(--lat)) calc(-1 * var(--offset)) var(--color),var(--lat) var(--offset) var(--color)}100%{box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}}@-webkit-keyframes loading-hash-after{0%{height:var(--thickness);box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}35%{height:var(--size);box-shadow:var(--offset) 0 var(--color),calc(-1 * var(--offset)) 0 var(--color)}70%{height:var(--thickness);box-shadow:var(--offset) calc(-1 * var(--lat)) var(--color),calc(-1 * var(--offset)) var(--lat) var(--color)}100%{box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}}@keyframes loading-hash-after{0%{height:var(--thickness);box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}35%{height:var(--size);box-shadow:var(--offset) 0 var(--color),calc(-1 * var(--offset)) 0 var(--color)}70%{height:var(--thickness);box-shadow:var(--offset) calc(-1 * var(--lat)) var(--color),calc(-1 * var(--offset)) var(--lat) var(--color)}100%{box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}}"]
                    }] }
        ];
        /** @nocollapse */
        HashLoaderComponent.ctorParameters = function () { return [
            { type: core.ElementRef }
        ]; };
        HashLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return HashLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        HashLoaderComponent.prototype.color;
        /** @type {?} */
        HashLoaderComponent.prototype.size;
        /** @type {?} */
        HashLoaderComponent.prototype.sizeUnit;
        /**
         * @type {?}
         * @private
         */
        HashLoaderComponent.prototype.rgba;
        /**
         * @type {?}
         * @private
         */
        HashLoaderComponent.prototype.thickness;
        /**
         * @type {?}
         * @private
         */
        HashLoaderComponent.prototype.elementRef;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var GridLoaderComponent = /** @class */ (function () {
        function GridLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @return {?}
         */
        GridLoaderComponent.prototype.createGridWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'width': "" + ("" + ((this.size * 3) + (this.margin * 6)) + this.sizeUnit),
                'font-size': '0'
            };
        };
        /**
         * @param {?} rand
         * @return {?}
         */
        GridLoaderComponent.prototype.createGridStyle = /**
         * @param {?} rand
         * @return {?}
         */
        function (rand) {
            return {
                'display': 'inline-block',
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': '100%',
                'animation-fill-mode': 'both',
                'animation': "loading-grid " + ((rand / 100) + 0.6) + "s " + ((rand / 100) - 0.2) + "s infinite ease"
            };
        };
        /**
         * @param {?} top
         * @return {?}
         */
        GridLoaderComponent.prototype.random = /**
         * @param {?} top
         * @return {?}
         */
        function (top) {
            return Math.random() * top;
        };
        GridLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-grid-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createGridWrapperStyle()\">\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-grid{0%{transform:scale(1)}50%{transform:scale(.5);opacity:.7}100%{transform:scale(1);opacity:1}}@keyframes loading-grid{0%{transform:scale(1)}50%{transform:scale(.5);opacity:.7}100%{transform:scale(1);opacity:1}}"]
                    }] }
        ];
        /** @nocollapse */
        GridLoaderComponent.ctorParameters = function () { return []; };
        GridLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return GridLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        GridLoaderComponent.prototype.color;
        /** @type {?} */
        GridLoaderComponent.prototype.size;
        /** @type {?} */
        GridLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        GridLoaderComponent.prototype.margin;
        /** @type {?} */
        GridLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var FadeLoaderComponent = /** @class */ (function () {
        function FadeLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.height = 15;
            this.heightUnit = 'px';
            this.width = 5;
            this.widthUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
            this.radius = 2;
            this.radiusUnit = 'px';
            this.radiusFactor = 20;
        }
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.ngOnInit = /**
         * @return {?}
         */
        function () {
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'font-size': '0',
                'top': this.radiusFactor + "px",
                'left': this.radiusFactor + "px",
                'width': this.radiusFactor * 3 + "px",
                'height': this.radiusFactor * 3 + "px"
            };
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleA = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(1), { 'top': this.radiusFactor + "px", 'left': '0' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleB = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(2), { 'top': this.createQuarter() + "px", 'left': this.createQuarter() + "px", 'transform': 'rotate(-45deg)' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleC = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(3), { 'top': '0', 'left': this.radiusFactor + "px", 'transform': 'rotate(90deg)' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleD = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(4), { 'top': -1 * this.createQuarter() + "px", 'left': this.createQuarter() + "px", 'transform': 'rotate(45deg)' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleE = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(5), { 'top': -1 * this.radiusFactor + "px", 'left': '0' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleF = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(6), { 'top': -1 * this.createQuarter() + "px", 'left': -1 * this.createQuarter() + "px", 'transform': 'rotate(-45deg)' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleG = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(7), { 'top': '0', 'left': -1 * this.radiusFactor + "px", 'transform': 'rotate(90deg)' });
        };
        /**
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyleH = /**
         * @return {?}
         */
        function () {
            return __assign({}, this.createFadeStyle(8), { 'top': this.createQuarter() + "px", 'left': -1 * this.createQuarter() + "px", 'transform': 'rotate(45deg)' });
        };
        /**
         * @private
         * @param {?} index
         * @return {?}
         */
        FadeLoaderComponent.prototype.createFadeStyle = /**
         * @private
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'position': 'absolute',
                'width': "" + ("" + this.width + this.widthUnit),
                'height': "" + ("" + this.height + this.heightUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'background-color': "" + this.color,
                'border-radius': "" + ("" + this.radius + this.radiusUnit),
                'transition': '2s',
                'animation-fill-mode': 'both',
                'animation': "loading-fade 1.2s " + index * 0.12 + "s infinite ease-in-out"
            };
        };
        /**
         * @private
         * @return {?}
         */
        FadeLoaderComponent.prototype.createQuarter = /**
         * @private
         * @return {?}
         */
        function () {
            return (this.radiusFactor / 2) + (this.radiusFactor / 5.5);
        };
        FadeLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-fade-loader',
                        template: "<div class=\"loading\" [ngStyle]=\"createFadeWrapperStyle()\">\r\n  <div [ngStyle]=\"createFadeStyleA()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleB()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleC()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleD()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleE()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleF()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleG()\"></div>\r\n  <div [ngStyle]=\"createFadeStyleH()\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-fade{50%{opacity:.3}100%{opacity:1}}@keyframes loading-fade{50%{opacity:.3}100%{opacity:1}}"]
                    }] }
        ];
        /** @nocollapse */
        FadeLoaderComponent.ctorParameters = function () { return []; };
        FadeLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            height: [{ type: core.Input }],
            heightUnit: [{ type: core.Input }],
            width: [{ type: core.Input }],
            widthUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }],
            radius: [{ type: core.Input }],
            radiusUnit: [{ type: core.Input }]
        };
        return FadeLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        FadeLoaderComponent.prototype.color;
        /** @type {?} */
        FadeLoaderComponent.prototype.height;
        /** @type {?} */
        FadeLoaderComponent.prototype.heightUnit;
        /** @type {?} */
        FadeLoaderComponent.prototype.width;
        /** @type {?} */
        FadeLoaderComponent.prototype.widthUnit;
        /** @type {?} */
        FadeLoaderComponent.prototype.margin;
        /** @type {?} */
        FadeLoaderComponent.prototype.marginUnit;
        /** @type {?} */
        FadeLoaderComponent.prototype.radius;
        /** @type {?} */
        FadeLoaderComponent.prototype.radiusUnit;
        /**
         * @type {?}
         * @private
         */
        FadeLoaderComponent.prototype.radiusFactor;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var DotLoaderComponent = /** @class */ (function () {
        function DotLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 60;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        DotLoaderComponent.prototype.createDotWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'animation-fill-mode': 'forwards',
                'animation': 'loading-rotate 2s 0s infinite linear'
            };
        };
        /**
         * @param {?} index
         * @return {?}
         */
        DotLoaderComponent.prototype.createDotStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'position': 'absolute',
                'top': "" + (index % 2 ? '0' : 'auto'),
                'bottom': "" + (index % 2 ? 'auto' : '0'),
                'height': "" + ("" + this.size / 2 + this.sizeUnit),
                'width': "" + ("" + this.size / 2 + this.sizeUnit),
                'background-color': "" + this.color,
                'border-radius': '100%',
                'animation-fill-mode': 'forwards',
                'animation': "loading-bounce 2s " + (index === 2 ? '-1s' : '0s') + " infinite linear"
            };
        };
        DotLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-dot-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createDotWrapperStyle()\">\r\n  <div [ngStyle]=\"createDotStyle(1)\"></div>\r\n  <div [ngStyle]=\"createDotStyle(2)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-rotate{100%{transform:rotate(360deg)}}@keyframes loading-rotate{100%{transform:rotate(360deg)}}@-webkit-keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}@keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}"]
                    }] }
        ];
        /** @nocollapse */
        DotLoaderComponent.ctorParameters = function () { return []; };
        DotLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return DotLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        DotLoaderComponent.prototype.color;
        /** @type {?} */
        DotLoaderComponent.prototype.size;
        /** @type {?} */
        DotLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ClipLoaderComponent = /** @class */ (function () {
        function ClipLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 35;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        ClipLoaderComponent.prototype.createClipStyle = /**
         * @return {?}
         */
        function () {
            return {
                'background': 'transparent !important',
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'border-radius': '100%',
                'border': '2px solid',
                'border-color': "" + this.color,
                'border-bottom-color': 'transparent',
                'display': 'inline-block',
                'animation': ' loading-clip 0.75s 0s infinite linear',
                'animation-fill-mode': 'both'
            };
        };
        ClipLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-clip-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createClipStyle()\"></div>\r\n",
                        styles: ["@-webkit-keyframes loading-clip{0%{transform:rotate(0) scale(1)}50%{transform:rotate(180deg) scale(.8)}100%{transform:rotate(360deg) scale(1)}}@keyframes loading-clip{0%{transform:rotate(0) scale(1)}50%{transform:rotate(180deg) scale(.8)}100%{transform:rotate(360deg) scale(1)}}"]
                    }] }
        ];
        /** @nocollapse */
        ClipLoaderComponent.ctorParameters = function () { return []; };
        ClipLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return ClipLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        ClipLoaderComponent.prototype.color;
        /** @type {?} */
        ClipLoaderComponent.prototype.size;
        /** @type {?} */
        ClipLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var ClimbingBoxLoaderComponent = /** @class */ (function () {
        function ClimbingBoxLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        ClimbingBoxLoaderComponent.prototype.createClimbingBoxContainerStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': '7.1em',
                'height': '7.1em'
            };
        };
        /**
         * @return {?}
         */
        ClimbingBoxLoaderComponent.prototype.createClimbingBoxWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'absolute',
                'top': '50%',
                'left': '50%',
                'margin-top': '-2.7em',
                'margin-left': '-2.7em',
                'width': '5.4em',
                'height': '5.4em',
                'font-size': "" + ("" + this.size + this.sizeUnit)
            };
        };
        /**
         * @return {?}
         */
        ClimbingBoxLoaderComponent.prototype.createClimbingBoxStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'absolute',
                'left': '0',
                'bottom': '-0.1em',
                'height': '1em',
                'width': '1em',
                'background-color': 'transparent',
                'border-radius': '15%',
                'border': "0.25em solid " + this.color,
                'transform': 'translate(0, -1em) rotate(-45deg)',
                'animation-fill-mode': 'both',
                'animation': 'loading-climbing-box 2.5s infinite cubic-bezier(.79, 0, .47, .97)'
            };
        };
        /**
         * @return {?}
         */
        ClimbingBoxLoaderComponent.prototype.createHillStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'absolute',
                'width': '7.1em',
                'height': '7.1em',
                'top': '1.7em',
                'left': '1.7em',
                'border-left': "0.25em solid " + this.color,
                'transform': 'rotate(45deg)'
            };
        };
        ClimbingBoxLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-climbing-box-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createClimbingBoxContainerStyle()\">\r\n  <div [ngStyle]=\"createClimbingBoxWrapperStyle()\">\r\n    <div [ngStyle]=\"createClimbingBoxStyle()\"></div>\r\n    <div [ngStyle]=\"createHillStyle()\"></div>\r\n  </div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-climbing-box{0%{transform:translate(0,-1em) rotate(-45deg)}5%{transform:translate(0,-1em) rotate(-50deg)}20%{transform:translate(1em,-2em) rotate(47deg)}25%{transform:translate(1em,-2em) rotate(45deg)}30%{transform:translate(1em,-2em) rotate(40deg)}45%{transform:translate(2em,-3em) rotate(137deg)}50%{transform:translate(2em,-3em) rotate(135deg)}55%{transform:translate(2em,-3em) rotate(130deg)}70%{transform:translate(3em,-4em) rotate(217deg)}75%{transform:translate(3em,-4em) rotate(220deg)}100%{transform:translate(0,-1em) rotate(-225deg)}}@keyframes loading-climbing-box{0%{transform:translate(0,-1em) rotate(-45deg)}5%{transform:translate(0,-1em) rotate(-50deg)}20%{transform:translate(1em,-2em) rotate(47deg)}25%{transform:translate(1em,-2em) rotate(45deg)}30%{transform:translate(1em,-2em) rotate(40deg)}45%{transform:translate(2em,-3em) rotate(137deg)}50%{transform:translate(2em,-3em) rotate(135deg)}55%{transform:translate(2em,-3em) rotate(130deg)}70%{transform:translate(3em,-4em) rotate(217deg)}75%{transform:translate(3em,-4em) rotate(220deg)}100%{transform:translate(0,-1em) rotate(-225deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        ClimbingBoxLoaderComponent.ctorParameters = function () { return []; };
        ClimbingBoxLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return ClimbingBoxLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        ClimbingBoxLoaderComponent.prototype.color;
        /** @type {?} */
        ClimbingBoxLoaderComponent.prototype.size;
        /** @type {?} */
        ClimbingBoxLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var CircleLoaderComponent = /** @class */ (function () {
        function CircleLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 50;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        CircleLoaderComponent.prototype.createCircleWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit)
            };
        };
        /**
         * @param {?} index
         * @return {?}
         */
        CircleLoaderComponent.prototype.createCircleStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'position': 'absolute',
                'height': "" + ("" + this.size * (1 - (index / 10)) + this.sizeUnit),
                'width': "" + ("" + this.size * (1 - (index / 10)) + this.sizeUnit),
                'border': "1px solid " + this.color,
                'border-radius': '100%',
                'transition': '2s',
                'border-bottom': 'none',
                'border-right': 'none',
                'top': (index * 0.7 * 2.5) + "%",
                'left': (index * 0.35 * 2.5) + "%",
                'animation-fill-mode': '',
                'animation': "loading-circle 1s " + index * 0.2 + "s infinite linear"
            };
        };
        CircleLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-circle-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createCircleWrapperStyle()\">\r\n  <div [ngStyle]=\"createCircleStyle(0)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(1)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(2)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(3)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(4)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-circle{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}@keyframes loading-circle{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}"]
                    }] }
        ];
        /** @nocollapse */
        CircleLoaderComponent.ctorParameters = function () { return []; };
        CircleLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return CircleLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        CircleLoaderComponent.prototype.color;
        /** @type {?} */
        CircleLoaderComponent.prototype.size;
        /** @type {?} */
        CircleLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var BounceLoaderComponent = /** @class */ (function () {
        function BounceLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 60;
            this.sizeUnit = 'px';
        }
        /**
         * @return {?}
         */
        BounceLoaderComponent.prototype.createBounceWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit)
            };
        };
        /**
         * @param {?} index
         * @return {?}
         */
        BounceLoaderComponent.prototype.createBounceStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'position': 'absolute',
                'height': "" + ("" + this.size + this.sizeUnit),
                'width': "" + ("" + this.size + this.sizeUnit),
                'background-color': "" + this.color,
                'border-radius': '100%',
                'opacity': '0.6',
                'top': '0',
                'left': '0',
                'animation-fill-mode': 'both',
                'animation': "loading-bounce 2.1s " + (index === 1 ? '1s' : '0s') + " infinite ease-in-out"
            };
        };
        BounceLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-bounce-loader',
                        template: "<div class=\"loader\" [ngStyle]=\"createBounceWrapperStyle()\">\r\n  <div [ngStyle]=\"createBounceStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBounceStyle(2)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}@keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}"]
                    }] }
        ];
        /** @nocollapse */
        BounceLoaderComponent.ctorParameters = function () { return []; };
        BounceLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }]
        };
        return BounceLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        BounceLoaderComponent.prototype.color;
        /** @type {?} */
        BounceLoaderComponent.prototype.size;
        /** @type {?} */
        BounceLoaderComponent.prototype.sizeUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var BeatLoaderComponent = /** @class */ (function () {
        function BeatLoaderComponent() {
            this.color = 'rgb(54, 215, 183)';
            this.size = 15;
            this.sizeUnit = 'px';
            this.margin = 2;
            this.marginUnit = 'px';
        }
        /**
         * @param {?} index
         * @return {?}
         */
        BeatLoaderComponent.prototype.createBeatStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'display': 'inline-block',
                'background-color': "" + this.color,
                'width': "" + ("" + this.size + this.sizeUnit),
                'height': "" + ("" + this.size + this.sizeUnit),
                'margin': "" + ("" + this.margin + this.marginUnit),
                'border-radius': '100%',
                'animation': "loading-beat 0.7s " + (index % 2 ? '0s' : '0.35s') + " infinite linear",
                'animation-fill-mode': 'both'
            };
        };
        BeatLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-beat-loader',
                        template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createBeatStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBeatStyle(2)\"></div>\r\n  <div [ngStyle]=\"createBeatStyle(3)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-beat{50%{transform:scale(.75);opacity:.2}100%{transform:scale(1);opacity:1}}@keyframes loading-beat{50%{transform:scale(.75);opacity:.2}100%{transform:scale(1);opacity:1}}"]
                    }] }
        ];
        /** @nocollapse */
        BeatLoaderComponent.ctorParameters = function () { return []; };
        BeatLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            size: [{ type: core.Input }],
            sizeUnit: [{ type: core.Input }],
            margin: [{ type: core.Input }],
            marginUnit: [{ type: core.Input }]
        };
        return BeatLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        BeatLoaderComponent.prototype.color;
        /** @type {?} */
        BeatLoaderComponent.prototype.size;
        /** @type {?} */
        BeatLoaderComponent.prototype.sizeUnit;
        /** @type {?} */
        BeatLoaderComponent.prototype.margin;
        /** @type {?} */
        BeatLoaderComponent.prototype.marginUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    var BarLoaderComponent = /** @class */ (function () {
        function BarLoaderComponent() {
            this.color = '#36d7b7';
            this.width = 100;
            this.widthUnit = 'px';
            this.height = 4;
            this.heightUnit = 'px';
        }
        /**
         * @return {?}
         */
        BarLoaderComponent.prototype.createBarWrapperStyle = /**
         * @return {?}
         */
        function () {
            return {
                'position': 'relative',
                'width': "" + ("" + this.width + this.widthUnit),
                'height': "" + ("" + this.height + this.heightUnit),
                'overflow': 'hidden',
                'background-color': "" + calculateRgba(this.color, 0.2),
                'background-clip': 'padding-box',
            };
        };
        /**
         * @param {?} index
         * @return {?}
         */
        BarLoaderComponent.prototype.createBarStyle = /**
         * @param {?} index
         * @return {?}
         */
        function (index) {
            return {
                'display': 'block',
                'position': 'absolute',
                'height': "" + ("" + this.height + this.heightUnit),
                'overflow': 'hidden',
                'background-color': "" + this.color,
                'background-clip': 'padding-box',
                'border-radius': '2px',
                'will-change': 'left, right',
                'animation-fill-mode': 'forwards',
                'animation': (index === 1 ? 'loading-long' : 'loading-short') + " 2.1s " + (index === 2 ? '1.15s' : '') + "\n                    " + (index === 1 ? 'cubic-bezier(0.65, 0.815, 0.735, 0.395)' : 'cubic-bezier(0.165, 0.84, 0.44, 1)') + " infinite"
            };
        };
        BarLoaderComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ngx-bar-loader',
                        template: "<div class=\"loading\" [ngStyle]=\"createBarWrapperStyle()\">\r\n  <div [ngStyle]=\"createBarStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBarStyle(2)\"></div>\r\n</div>\r\n",
                        styles: ["@-webkit-keyframes loading-long{0%{left:-35%;right:100%}100%,60%{left:100%;right:-90%}}@keyframes loading-long{0%{left:-35%;right:100%}100%,60%{left:100%;right:-90%}}@-webkit-keyframes loading-short{0%{left:-200%;right:100%}100%,60%{left:107%;right:-8%}}@keyframes loading-short{0%{left:-200%;right:100%}100%,60%{left:107%;right:-8%}}"]
                    }] }
        ];
        /** @nocollapse */
        BarLoaderComponent.ctorParameters = function () { return []; };
        BarLoaderComponent.propDecorators = {
            color: [{ type: core.Input }],
            width: [{ type: core.Input }],
            widthUnit: [{ type: core.Input }],
            height: [{ type: core.Input }],
            heightUnit: [{ type: core.Input }]
        };
        return BarLoaderComponent;
    }());
    if (false) {
        /** @type {?} */
        BarLoaderComponent.prototype.color;
        /** @type {?} */
        BarLoaderComponent.prototype.width;
        /** @type {?} */
        BarLoaderComponent.prototype.widthUnit;
        /** @type {?} */
        BarLoaderComponent.prototype.height;
        /** @type {?} */
        BarLoaderComponent.prototype.heightUnit;
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
     */
    /** @type {?} */
    var components = [
        SyncLoaderComponent,
        SquareLoaderComponent,
        SkewLoaderComponent,
        ScaleLoaderComponent,
        RotateLoaderComponent,
        RiseLoaderComponent,
        RingLoaderComponent,
        PulseLoaderComponent,
        PropagateLoaderComponent,
        PacmanLoaderComponent,
        MoonLoaderComponent,
        HashLoaderComponent,
        GridLoaderComponent,
        FadeLoaderComponent,
        DotLoaderComponent,
        ClipLoaderComponent,
        ClimbingBoxLoaderComponent,
        CircleLoaderComponent,
        BounceLoaderComponent,
        BeatLoaderComponent,
        BarLoaderComponent
    ];
    var NgxSpinnersModule = /** @class */ (function () {
        function NgxSpinnersModule() {
        }
        NgxSpinnersModule.decorators = [
            { type: core.NgModule, args: [{
                        declarations: components,
                        imports: [common.CommonModule],
                        exports: components
                    },] }
        ];
        return NgxSpinnersModule;
    }());

    exports.NgxSpinnersModule = NgxSpinnersModule;
    exports.a = SyncLoaderComponent;
    exports.b = SquareLoaderComponent;
    exports.c = SkewLoaderComponent;
    exports.d = ScaleLoaderComponent;
    exports.e = RotateLoaderComponent;
    exports.f = RiseLoaderComponent;
    exports.g = RingLoaderComponent;
    exports.h = PulseLoaderComponent;
    exports.i = PropagateLoaderComponent;
    exports.j = PacmanLoaderComponent;
    exports.k = MoonLoaderComponent;
    exports.l = HashLoaderComponent;
    exports.m = GridLoaderComponent;
    exports.n = FadeLoaderComponent;
    exports.o = DotLoaderComponent;
    exports.p = ClipLoaderComponent;
    exports.q = ClimbingBoxLoaderComponent;
    exports.r = CircleLoaderComponent;
    exports.s = BounceLoaderComponent;
    exports.t = BeatLoaderComponent;
    exports.u = BarLoaderComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
