// ngx-spinners/esm2015/ngx-spinners.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/**
 * Generated bundle index. Do not edit.
 */
export { NgxSpinnersModule } from './public_api';
export { BarLoaderComponent as ɵu } from './lib/components/bar-loader.component';
export { BeatLoaderComponent as ɵt } from './lib/components/beat-loader.component';
export { BounceLoaderComponent as ɵs } from './lib/components/bounce-loader.component';
export { CircleLoaderComponent as ɵr } from './lib/components/circle-loader.component';
export { ClimbingBoxLoaderComponent as ɵq } from './lib/components/climbing-box-loader.component';
export { ClipLoaderComponent as ɵp } from './lib/components/clip-loader.component';
export { DotLoaderComponent as ɵo } from './lib/components/dot-loader.component';
export { FadeLoaderComponent as ɵn } from './lib/components/fade-loader.component';
export { GridLoaderComponent as ɵm } from './lib/components/grid-loader.component';
export { HashLoaderComponent as ɵl } from './lib/components/hash-loader.component';
export { MoonLoaderComponent as ɵk } from './lib/components/moon-loader.component';
export { PacmanLoaderComponent as ɵj } from './lib/components/pacman-loader.component';
export { PropagateLoaderComponent as ɵi } from './lib/components/propagate-loader.component';
export { PulseLoaderComponent as ɵh } from './lib/components/pulse-loader.component';
export { RingLoaderComponent as ɵg } from './lib/components/ring-loader.component';
export { RiseLoaderComponent as ɵf } from './lib/components/rise-loader.component';
export { RotateLoaderComponent as ɵe } from './lib/components/rotate-loader.component';
export { ScaleLoaderComponent as ɵd } from './lib/components/scale-loader.component';
export { SkewLoaderComponent as ɵc } from './lib/components/skew-loader.component';
export { SquareLoaderComponent as ɵb } from './lib/components/square-loader.component';
export { SyncLoaderComponent as ɵa } from './lib/components/sync-loader.component';
