// ngx-spinners/esm5/lib/components/beat-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var BeatLoaderComponent = /** @class */ (function () {
    function BeatLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @param {?} index
     * @return {?}
     */
    BeatLoaderComponent.prototype.createBeatStyle = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return {
            'display': 'inline-block',
            'background-color': "" + this.color,
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'margin': "" + ("" + this.margin + this.marginUnit),
            'border-radius': '100%',
            'animation': "loading-beat 0.7s " + (index % 2 ? '0s' : '0.35s') + " infinite linear",
            'animation-fill-mode': 'both'
        };
    };
    BeatLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-beat-loader',
                    template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createBeatStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBeatStyle(2)\"></div>\r\n  <div [ngStyle]=\"createBeatStyle(3)\"></div>\r\n</div>\r\n",
                    styles: ["@-webkit-keyframes loading-beat{50%{transform:scale(.75);opacity:.2}100%{transform:scale(1);opacity:1}}@keyframes loading-beat{50%{transform:scale(.75);opacity:.2}100%{transform:scale(1);opacity:1}}"]
                }] }
    ];
    /** @nocollapse */
    BeatLoaderComponent.ctorParameters = function () { return []; };
    BeatLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }],
        margin: [{ type: Input }],
        marginUnit: [{ type: Input }]
    };
    return BeatLoaderComponent;
}());
export { BeatLoaderComponent };
if (false) {
    /** @type {?} */
    BeatLoaderComponent.prototype.color;
    /** @type {?} */
    BeatLoaderComponent.prototype.size;
    /** @type {?} */
    BeatLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    BeatLoaderComponent.prototype.margin;
    /** @type {?} */
    BeatLoaderComponent.prototype.marginUnit;
}
