// ngx-spinners/esm5/lib/components/climbing-box-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var ClimbingBoxLoaderComponent = /** @class */ (function () {
    function ClimbingBoxLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    ClimbingBoxLoaderComponent.prototype.createClimbingBoxContainerStyle = /**
     * @return {?}
     */
    function () {
        return {
            'position': 'relative',
            'width': '7.1em',
            'height': '7.1em'
        };
    };
    /**
     * @return {?}
     */
    ClimbingBoxLoaderComponent.prototype.createClimbingBoxWrapperStyle = /**
     * @return {?}
     */
    function () {
        return {
            'position': 'absolute',
            'top': '50%',
            'left': '50%',
            'margin-top': '-2.7em',
            'margin-left': '-2.7em',
            'width': '5.4em',
            'height': '5.4em',
            'font-size': "" + ("" + this.size + this.sizeUnit)
        };
    };
    /**
     * @return {?}
     */
    ClimbingBoxLoaderComponent.prototype.createClimbingBoxStyle = /**
     * @return {?}
     */
    function () {
        return {
            'position': 'absolute',
            'left': '0',
            'bottom': '-0.1em',
            'height': '1em',
            'width': '1em',
            'background-color': 'transparent',
            'border-radius': '15%',
            'border': "0.25em solid " + this.color,
            'transform': 'translate(0, -1em) rotate(-45deg)',
            'animation-fill-mode': 'both',
            'animation': 'loading-climbing-box 2.5s infinite cubic-bezier(.79, 0, .47, .97)'
        };
    };
    /**
     * @return {?}
     */
    ClimbingBoxLoaderComponent.prototype.createHillStyle = /**
     * @return {?}
     */
    function () {
        return {
            'position': 'absolute',
            'width': '7.1em',
            'height': '7.1em',
            'top': '1.7em',
            'left': '1.7em',
            'border-left': "0.25em solid " + this.color,
            'transform': 'rotate(45deg)'
        };
    };
    ClimbingBoxLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-climbing-box-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createClimbingBoxContainerStyle()\">\r\n  <div [ngStyle]=\"createClimbingBoxWrapperStyle()\">\r\n    <div [ngStyle]=\"createClimbingBoxStyle()\"></div>\r\n    <div [ngStyle]=\"createHillStyle()\"></div>\r\n  </div>\r\n</div>\r\n",
                    styles: ["@-webkit-keyframes loading-climbing-box{0%{transform:translate(0,-1em) rotate(-45deg)}5%{transform:translate(0,-1em) rotate(-50deg)}20%{transform:translate(1em,-2em) rotate(47deg)}25%{transform:translate(1em,-2em) rotate(45deg)}30%{transform:translate(1em,-2em) rotate(40deg)}45%{transform:translate(2em,-3em) rotate(137deg)}50%{transform:translate(2em,-3em) rotate(135deg)}55%{transform:translate(2em,-3em) rotate(130deg)}70%{transform:translate(3em,-4em) rotate(217deg)}75%{transform:translate(3em,-4em) rotate(220deg)}100%{transform:translate(0,-1em) rotate(-225deg)}}@keyframes loading-climbing-box{0%{transform:translate(0,-1em) rotate(-45deg)}5%{transform:translate(0,-1em) rotate(-50deg)}20%{transform:translate(1em,-2em) rotate(47deg)}25%{transform:translate(1em,-2em) rotate(45deg)}30%{transform:translate(1em,-2em) rotate(40deg)}45%{transform:translate(2em,-3em) rotate(137deg)}50%{transform:translate(2em,-3em) rotate(135deg)}55%{transform:translate(2em,-3em) rotate(130deg)}70%{transform:translate(3em,-4em) rotate(217deg)}75%{transform:translate(3em,-4em) rotate(220deg)}100%{transform:translate(0,-1em) rotate(-225deg)}}"]
                }] }
    ];
    /** @nocollapse */
    ClimbingBoxLoaderComponent.ctorParameters = function () { return []; };
    ClimbingBoxLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }]
    };
    return ClimbingBoxLoaderComponent;
}());
export { ClimbingBoxLoaderComponent };
if (false) {
    /** @type {?} */
    ClimbingBoxLoaderComponent.prototype.color;
    /** @type {?} */
    ClimbingBoxLoaderComponent.prototype.size;
    /** @type {?} */
    ClimbingBoxLoaderComponent.prototype.sizeUnit;
}
