// ngx-spinners/esm5/lib/components/grid-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var GridLoaderComponent = /** @class */ (function () {
    function GridLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @return {?}
     */
    GridLoaderComponent.prototype.createGridWrapperStyle = /**
     * @return {?}
     */
    function () {
        return {
            'width': "" + ("" + ((this.size * 3) + (this.margin * 6)) + this.sizeUnit),
            'font-size': '0'
        };
    };
    /**
     * @param {?} rand
     * @return {?}
     */
    GridLoaderComponent.prototype.createGridStyle = /**
     * @param {?} rand
     * @return {?}
     */
    function (rand) {
        return {
            'display': 'inline-block',
            'background-color': "" + this.color,
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'margin': "" + ("" + this.margin + this.marginUnit),
            'border-radius': '100%',
            'animation-fill-mode': 'both',
            'animation': "loading-grid " + ((rand / 100) + 0.6) + "s " + ((rand / 100) - 0.2) + "s infinite ease"
        };
    };
    /**
     * @param {?} top
     * @return {?}
     */
    GridLoaderComponent.prototype.random = /**
     * @param {?} top
     * @return {?}
     */
    function (top) {
        return Math.random() * top;
    };
    GridLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-grid-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createGridWrapperStyle()\">\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n  <div [ngStyle]=\"createGridStyle(random(100))\"></div>\r\n</div>\r\n",
                    styles: ["@-webkit-keyframes loading-grid{0%{transform:scale(1)}50%{transform:scale(.5);opacity:.7}100%{transform:scale(1);opacity:1}}@keyframes loading-grid{0%{transform:scale(1)}50%{transform:scale(.5);opacity:.7}100%{transform:scale(1);opacity:1}}"]
                }] }
    ];
    /** @nocollapse */
    GridLoaderComponent.ctorParameters = function () { return []; };
    GridLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }],
        margin: [{ type: Input }],
        marginUnit: [{ type: Input }]
    };
    return GridLoaderComponent;
}());
export { GridLoaderComponent };
if (false) {
    /** @type {?} */
    GridLoaderComponent.prototype.color;
    /** @type {?} */
    GridLoaderComponent.prototype.size;
    /** @type {?} */
    GridLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    GridLoaderComponent.prototype.margin;
    /** @type {?} */
    GridLoaderComponent.prototype.marginUnit;
}
