// ngx-spinners/esm5/lib/components/hash-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input } from '@angular/core';
import { calculateRgba } from '../helpers/helpers';
var HashLoaderComponent = /** @class */ (function () {
    function HashLoaderComponent(elementRef) {
        this.elementRef = elementRef;
        this.color = '#36d7b7';
        this.size = 50;
        this.sizeUnit = 'px';
        this.rgba = (/**
         * @param {?} color
         * @return {?}
         */
        function (color) { return calculateRgba(color, 0.75); });
        this.thickness = (/**
         * @param {?} size
         * @return {?}
         */
        function (size) { return size / 5; });
    }
    /**
     * @return {?}
     */
    HashLoaderComponent.prototype.createHashWrapperStyle = /**
     * @return {?}
     */
    function () {
        return {
            'position': 'relative',
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'transform': 'rotate(165deg)'
        };
    };
    /**
     * @param {?} index
     * @param {?} color
     * @param {?} size
     * @param {?} sizeUnit
     * @return {?}
     */
    HashLoaderComponent.prototype.createHashStyle = /**
     * @param {?} index
     * @param {?} color
     * @param {?} size
     * @param {?} sizeUnit
     * @return {?}
     */
    function (index, color, size, sizeUnit) {
        this.setStyleVariables(color, size, sizeUnit);
        return {
            'display': 'block',
            'position': 'absolute',
            'content': '',
            'top': '50%',
            'left': '50%',
            'width': "" + ("" + this.size / 5 + this.sizeUnit),
            'height': "" + ("" + this.size / 5 + this.sizeUnit),
            'border-radius': "" + ("" + this.size / 10 + this.sizeUnit),
            'transform': 'translate(-50%, -50%)',
            'animation-fill-mode': 'none',
            'animation': (index === 1 ? 'loading-hash-before' : 'loading-hash-after') + " 2s infinite"
        };
    };
    /**
     * @private
     * @param {?} size
     * @return {?}
     */
    HashLoaderComponent.prototype.lat = /**
     * @private
     * @param {?} size
     * @return {?}
     */
    function (size) {
        return (this.size - this.thickness(size)) / 2;
    };
    /**
     * @private
     * @param {?} size
     * @return {?}
     */
    HashLoaderComponent.prototype.offset = /**
     * @private
     * @param {?} size
     * @return {?}
     */
    function (size) {
        return this.lat(size) - this.thickness(size);
    };
    /**
     * @private
     * @param {?} color
     * @param {?} size
     * @param {?} sizeUnit
     * @return {?}
     */
    HashLoaderComponent.prototype.setStyleVariables = /**
     * @private
     * @param {?} color
     * @param {?} size
     * @param {?} sizeUnit
     * @return {?}
     */
    function (color, size, sizeUnit) {
        /** @type {?} */
        var rgba = this.rgba(color);
        /** @type {?} */
        var lat = this.lat(size);
        /** @type {?} */
        var thickness = this.thickness(size);
        /** @type {?} */
        var offset = this.offset(size);
        this.elementRef.nativeElement.style.setProperty('--color', rgba);
        this.elementRef.nativeElement.style.setProperty('--size', "" + ("" + size + sizeUnit));
        this.elementRef.nativeElement.style.setProperty('--lat', "" + (lat + "px"));
        this.elementRef.nativeElement.style.setProperty('--thickness', "" + (thickness + "px"));
        this.elementRef.nativeElement.style.setProperty('--offset', "" + (offset + "px"));
    };
    HashLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-hash-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createHashWrapperStyle()\">\r\n      <div [ngStyle]=\"createHashStyle(1,color, size, sizeUnit)\"></div>\r\n      <div [ngStyle]=\"createHashStyle(2, color, size, sizeUnit)\"></div>\r\n</div>\r\n",
                    styles: [":host{--color:rgba(54, 215, 183, 0.75);--thickness:10px;--size:50px;--lat:20px;--offset:10px}@-webkit-keyframes loading-hash-before{0%{width:var(--thickness);box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}35%{width:var(--size);box-shadow:0 calc(-1 * var(--offset)) var(--color),0 var(--offset) var(--color)}70%{width:var(--thickness);box-shadow:calc(-1 * var(--lat)) calc(-1 * var(--offset)) var(--color),var(--lat) var(--offset) var(--color)}100%{box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}}@keyframes loading-hash-before{0%{width:var(--thickness);box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}35%{width:var(--size);box-shadow:0 calc(-1 * var(--offset)) var(--color),0 var(--offset) var(--color)}70%{width:var(--thickness);box-shadow:calc(-1 * var(--lat)) calc(-1 * var(--offset)) var(--color),var(--lat) var(--offset) var(--color)}100%{box-shadow:var(--lat) calc(-1 * var(--offset)) var(--color),calc(-1 * var(--lat)) var(--offset) var(--color)}}@-webkit-keyframes loading-hash-after{0%{height:var(--thickness);box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}35%{height:var(--size);box-shadow:var(--offset) 0 var(--color),calc(-1 * var(--offset)) 0 var(--color)}70%{height:var(--thickness);box-shadow:var(--offset) calc(-1 * var(--lat)) var(--color),calc(-1 * var(--offset)) var(--lat) var(--color)}100%{box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}}@keyframes loading-hash-after{0%{height:var(--thickness);box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}35%{height:var(--size);box-shadow:var(--offset) 0 var(--color),calc(-1 * var(--offset)) 0 var(--color)}70%{height:var(--thickness);box-shadow:var(--offset) calc(-1 * var(--lat)) var(--color),calc(-1 * var(--offset)) var(--lat) var(--color)}100%{box-shadow:var(--offset) var(--lat) var(--color),calc(-1 * var(--offset)) calc(-1 * var(--lat)) var(--color)}}"]
                }] }
    ];
    /** @nocollapse */
    HashLoaderComponent.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    HashLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }]
    };
    return HashLoaderComponent;
}());
export { HashLoaderComponent };
if (false) {
    /** @type {?} */
    HashLoaderComponent.prototype.color;
    /** @type {?} */
    HashLoaderComponent.prototype.size;
    /** @type {?} */
    HashLoaderComponent.prototype.sizeUnit;
    /**
     * @type {?}
     * @private
     */
    HashLoaderComponent.prototype.rgba;
    /**
     * @type {?}
     * @private
     */
    HashLoaderComponent.prototype.thickness;
    /**
     * @type {?}
     * @private
     */
    HashLoaderComponent.prototype.elementRef;
}
