// ngx-spinners/esm5/lib/components/propagate-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var PropagateLoaderComponent = /** @class */ (function () {
    function PropagateLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    PropagateLoaderComponent.prototype.createPropagateWrapperStyle = /**
     * @return {?}
     */
    function () {
        return {
            'position': 'relative'
        };
    };
    /**
     * @param {?} index
     * @return {?}
     */
    PropagateLoaderComponent.prototype.createPropagateStyle = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return {
            'position': 'absolute',
            'font-size': "" + ("" + this.size / 3 + this.sizeUnit),
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'background': "" + this.color,
            'border-radius': '50%',
            'animation': 'loading-propagate-' + index + " 1.5s  infinite",
            'animation-fill-mode': 'forwards'
        };
    };
    PropagateLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-propagate-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createPropagateWrapperStyle()\">\r\n  <div [ngStyle]=\"createPropagateStyle(0)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(1)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(2)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(3)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(4)\"></div>\r\n  <div [ngStyle]=\"createPropagateStyle(5)\"></div>\r\n</div>\r\n",
                    styles: [":host{--oneRem:1rem;--threeRem:3rem;--fiveRem:5rem}@-webkit-keyframes loading-propagate-0{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}75%{transform:translateX(calc(-1 * var(--fiveRem))) scale(.5)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-0{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}75%{transform:translateX(calc(-1 * var(--fiveRem))) scale(.5)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-1{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%,75%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-1{25%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}50%,75%{transform:translateX(calc(-1 * var(--threeRem))) scale(.6)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-2{25%,75%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-2{25%,75%{transform:translateX(calc(-1 * var(--oneRem))) scale(.75)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-3{25%,75%{transform:translateX(var(--oneRem)) scale(.75)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-3{25%,75%{transform:translateX(var(--oneRem)) scale(.75)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-4{25%{transform:translateX(var(--oneRem)) scale(.75)}50%,75%{transform:translateX(var(--threeRem)) scale(.6)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-4{25%{transform:translateX(var(--oneRem)) scale(.75)}50%,75%{transform:translateX(var(--threeRem)) scale(.6)}95%{transform:translateX(0) scale(1)}}@-webkit-keyframes loading-propagate-5{25%{transform:translateX(var(--oneRem)) scale(.75)}50%{transform:translateX(var(--threeRem)) scale(.6)}75%{transform:translateX(var(--fiveRem)) scale(.5)}95%{transform:translateX(0) scale(1)}}@keyframes loading-propagate-5{25%{transform:translateX(var(--oneRem)) scale(.75)}50%{transform:translateX(var(--threeRem)) scale(.6)}75%{transform:translateX(var(--fiveRem)) scale(.5)}95%{transform:translateX(0) scale(1)}}"]
                }] }
    ];
    /** @nocollapse */
    PropagateLoaderComponent.ctorParameters = function () { return []; };
    PropagateLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }]
    };
    return PropagateLoaderComponent;
}());
export { PropagateLoaderComponent };
if (false) {
    /** @type {?} */
    PropagateLoaderComponent.prototype.color;
    /** @type {?} */
    PropagateLoaderComponent.prototype.size;
    /** @type {?} */
    PropagateLoaderComponent.prototype.sizeUnit;
}
