// ngx-spinners/esm5/lib/components/ring-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var RingLoaderComponent = /** @class */ (function () {
    function RingLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 60;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    RingLoaderComponent.prototype.createRingWrapperStyle = /**
     * @return {?}
     */
    function () {
        return {
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'position': 'relative'
        };
    };
    /**
     * @param {?} index
     * @return {?}
     */
    RingLoaderComponent.prototype.createRingStyle = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return {
            'position': 'absolute',
            'top': '0',
            'left': '0',
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'border': "" + this.size / 10 + this.sizeUnit + " solid " + this.color,
            'opacity': '0.4',
            'border-radius': '100%',
            'animation-fill-mode': 'forwards',
            'perspective': '800px',
            'animation': (index === 1 ? 'loading-ring-right' : 'loading-ring-left') + " 2s 0s infinite linear"
        };
    };
    RingLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-ring-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createRingWrapperStyle()\">\r\n  <div [ngStyle]=\"createRingStyle(1)\"></div>\r\n  <div [ngStyle]=\"createRingStyle(2)\"></div>\r\n</div>\r\n",
                    styles: ["@-webkit-keyframes loading-ring-right{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(180deg) rotateY(360deg) rotateZ(360deg)}}@keyframes loading-ring-right{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(180deg) rotateY(360deg) rotateZ(360deg)}}@-webkit-keyframes loading-ring-left{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(360deg) rotateY(180deg) rotateZ(360deg)}}@keyframes loading-ring-left{0%{transform:rotateX(0) rotateY(0) rotateZ(0)}100%{transform:rotateX(360deg) rotateY(180deg) rotateZ(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    RingLoaderComponent.ctorParameters = function () { return []; };
    RingLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }]
    };
    return RingLoaderComponent;
}());
export { RingLoaderComponent };
if (false) {
    /** @type {?} */
    RingLoaderComponent.prototype.color;
    /** @type {?} */
    RingLoaderComponent.prototype.size;
    /** @type {?} */
    RingLoaderComponent.prototype.sizeUnit;
}
