// ngx-spinners/esm5/lib/components/rotate-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input, ViewEncapsulation } from '@angular/core';
var RotateLoaderComponent = /** @class */ (function () {
    function RotateLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @return {?}
     */
    RotateLoaderComponent.prototype.createRotateWrapperStyle = /**
     * @return {?}
     */
    function () {
        return tslib_1.__assign({ 'display': 'inline-block' }, this.createBallStyle(), { 'position': 'relative', 'animation-fill-mode': 'both', 'animation': 'loading-rotate 1s 0s infinite cubic-bezier(.7,-.13,.22,.86)' });
    };
    /**
     * @return {?}
     */
    RotateLoaderComponent.prototype.createRotateLongStyle = /**
     * @return {?}
     */
    function () {
        return tslib_1.__assign({}, this.createBallStyle(), this.createPositionStyle(1));
    };
    /**
     * @return {?}
     */
    RotateLoaderComponent.prototype.createRotateShortStyle = /**
     * @return {?}
     */
    function () {
        return tslib_1.__assign({}, this.createBallStyle(), this.createPositionStyle(2));
    };
    /**
     * @private
     * @return {?}
     */
    RotateLoaderComponent.prototype.createBallStyle = /**
     * @private
     * @return {?}
     */
    function () {
        return {
            'background-color': "" + this.color,
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'margin': "" + ("" + this.margin + this.marginUnit),
            'border-radius': '100%'
        };
    };
    /**
     * @private
     * @param {?} index
     * @return {?}
     */
    RotateLoaderComponent.prototype.createPositionStyle = /**
     * @private
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return {
            'opacity': '0.8',
            'position': 'absolute',
            'top': '0',
            'left': (index % 2 ? -28 : 25) + "px"
        };
    };
    RotateLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-rotate-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createRotateWrapperStyle()\">\r\n  <div [ngStyle]=\"createRotateLongStyle()\"></div>\r\n  <div [ngStyle]=\"createRotateShortStyle()\"></div>\r\n</div>\r\n",
                    encapsulation: ViewEncapsulation.ShadowDom,
                    styles: ["@-webkit-keyframes loading-rotate{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}@keyframes loading-rotate{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    RotateLoaderComponent.ctorParameters = function () { return []; };
    RotateLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }],
        margin: [{ type: Input }],
        marginUnit: [{ type: Input }]
    };
    return RotateLoaderComponent;
}());
export { RotateLoaderComponent };
if (false) {
    /** @type {?} */
    RotateLoaderComponent.prototype.color;
    /** @type {?} */
    RotateLoaderComponent.prototype.size;
    /** @type {?} */
    RotateLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    RotateLoaderComponent.prototype.margin;
    /** @type {?} */
    RotateLoaderComponent.prototype.marginUnit;
}
