// ngx-spinners/esm5/lib/components/scale-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var ScaleLoaderComponent = /** @class */ (function () {
    function ScaleLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.width = 4;
        this.widthUnit = 'px';
        this.height = 35;
        this.heightUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
        this.radius = 2;
        this.radiusUnit = 'px';
    }
    /**
     * @param {?} index
     * @return {?}
     */
    ScaleLoaderComponent.prototype.createScaleStyle = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return {
            'display': 'inline-block',
            'background-color': "" + this.color,
            'width': "" + ("" + this.width + this.widthUnit),
            'height': "" + ("" + this.height + this.heightUnit),
            'margin': "" + ("" + this.margin + this.marginUnit),
            'border-radius': "" + ("" + this.radius + this.radiusUnit),
            'animation': "loading-scale 1s " + index * 0.1 + "s infinite cubic-bezier(.2,.68,.18,1.08)",
            'animation-fill-mode': 'both'
        };
    };
    ScaleLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-scale-loader',
                    template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createScaleStyle(1)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(2)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(3)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(4)\"></div>\r\n  <div [ngStyle]=\"createScaleStyle(5)\"></div>\r\n</div>\r\n",
                    styles: ["@-webkit-keyframes loading-scale{0%,100%{transform:scaleY(1)}50%{transform:scaleY(.4)}}@keyframes loading-scale{0%,100%{transform:scaleY(1)}50%{transform:scaleY(.4)}}"]
                }] }
    ];
    /** @nocollapse */
    ScaleLoaderComponent.ctorParameters = function () { return []; };
    ScaleLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        width: [{ type: Input }],
        widthUnit: [{ type: Input }],
        height: [{ type: Input }],
        heightUnit: [{ type: Input }],
        margin: [{ type: Input }],
        marginUnit: [{ type: Input }],
        radius: [{ type: Input }],
        radiusUnit: [{ type: Input }]
    };
    return ScaleLoaderComponent;
}());
export { ScaleLoaderComponent };
if (false) {
    /** @type {?} */
    ScaleLoaderComponent.prototype.color;
    /** @type {?} */
    ScaleLoaderComponent.prototype.width;
    /** @type {?} */
    ScaleLoaderComponent.prototype.widthUnit;
    /** @type {?} */
    ScaleLoaderComponent.prototype.height;
    /** @type {?} */
    ScaleLoaderComponent.prototype.heightUnit;
    /** @type {?} */
    ScaleLoaderComponent.prototype.margin;
    /** @type {?} */
    ScaleLoaderComponent.prototype.marginUnit;
    /** @type {?} */
    ScaleLoaderComponent.prototype.radius;
    /** @type {?} */
    ScaleLoaderComponent.prototype.radiusUnit;
}
