// ngx-spinners/esm5/lib/components/skew-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var SkewLoaderComponent = /** @class */ (function () {
    function SkewLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    SkewLoaderComponent.prototype.createSkewStyle = /**
     * @return {?}
     */
    function () {
        return {
            'display': 'inline-block',
            'width': '0',
            'height': '0',
            'border-left': "" + this.size + this.sizeUnit + " solid transparent",
            'border-right': "" + this.size + this.sizeUnit + " solid transparent",
            'border-bottom': "" + this.size + this.sizeUnit + " solid " + this.color,
            'animation': 'loading-skew 3s 0s infinite cubic-bezier(.09,.57,.49,.9)',
            'animation-fill-mode': 'both'
        };
    };
    SkewLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-skew-loader',
                    template: "<div class=\"loader\" [ngStyle]=\"createSkewStyle()\"></div>\r\n",
                    styles: ["@-webkit-keyframes loading-skew{25%{transform:perspective(100px) rotateX(180deg) rotateY(0)}50%{transform:perspective(100px) rotateX(180deg) rotateY(180deg)}75%{transform:perspective(100px) rotateX(0) rotateY(180deg)}100%{transform:perspective(100px) rotateX(0) rotateY(0)}}@keyframes loading-skew{25%{transform:perspective(100px) rotateX(180deg) rotateY(0)}50%{transform:perspective(100px) rotateX(180deg) rotateY(180deg)}75%{transform:perspective(100px) rotateX(0) rotateY(180deg)}100%{transform:perspective(100px) rotateX(0) rotateY(0)}}"]
                }] }
    ];
    /** @nocollapse */
    SkewLoaderComponent.ctorParameters = function () { return []; };
    SkewLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }]
    };
    return SkewLoaderComponent;
}());
export { SkewLoaderComponent };
if (false) {
    /** @type {?} */
    SkewLoaderComponent.prototype.color;
    /** @type {?} */
    SkewLoaderComponent.prototype.size;
    /** @type {?} */
    SkewLoaderComponent.prototype.sizeUnit;
}
