// ngx-spinners/esm5/lib/components/sync-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
var SyncLoaderComponent = /** @class */ (function () {
    function SyncLoaderComponent() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @param {?} index
     * @return {?}
     */
    SyncLoaderComponent.prototype.createBallStyle = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return {
            'display': 'inline-block',
            'background-color': "" + this.color,
            'width': "" + ("" + this.size + this.sizeUnit),
            'height': "" + ("" + this.size + this.sizeUnit),
            'margin': "" + ("" + this.margin + this.marginUnit),
            'border-radius': '100%',
            'animation': "loading-sync 0.6s " + index * 0.07 + "s infinite ease-in-out",
            'animation-fill-mode': 'both'
        };
    };
    SyncLoaderComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ngx-sync-loader',
                    template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createBallStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(2)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(3)\"></div>\r\n</div>\r\n",
                    styles: ["@-webkit-keyframes loading-sync{33%{transform:translateY(10px)}66%{transform:translateY(-10px)}100%{transform:translateY(0)}}@keyframes loading-sync{33%{transform:translateY(10px)}66%{transform:translateY(-10px)}100%{transform:translateY(0)}}"]
                }] }
    ];
    /** @nocollapse */
    SyncLoaderComponent.ctorParameters = function () { return []; };
    SyncLoaderComponent.propDecorators = {
        color: [{ type: Input }],
        size: [{ type: Input }],
        sizeUnit: [{ type: Input }],
        margin: [{ type: Input }],
        marginUnit: [{ type: Input }]
    };
    return SyncLoaderComponent;
}());
export { SyncLoaderComponent };
if (false) {
    /** @type {?} */
    SyncLoaderComponent.prototype.color;
    /** @type {?} */
    SyncLoaderComponent.prototype.size;
    /** @type {?} */
    SyncLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    SyncLoaderComponent.prototype.margin;
    /** @type {?} */
    SyncLoaderComponent.prototype.marginUnit;
}
