// ngx-alerts/esm5/lib/service/alert.service.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Inject, Injectable } from '@angular/core';
import { BehaviorSubject, Subject, timer } from 'rxjs';
import { ALERT_CONFIG } from '../alert.config';
import { scan, take } from 'rxjs/internal/operators';
import { AlertReducer } from './alert.reducer';
var AlertService = /** @class */ (function () {
    function AlertService(config) {
        this.config = config;
        this.dispatcher = new Subject();
        this.state = new BehaviorSubject([]);
        this.initConfig();
        this.dispatcher
            .pipe(scan(AlertReducer.reduce, []))
            .subscribe(this.state);
    }
    /**
     * @private
     * @return {?}
     */
    AlertService.prototype.initConfig = /**
     * @private
     * @return {?}
     */
    function () {
        if (!this.config) {
            this.config = {};
        }
        this.config.timeout = !!this.config.timeout ? this.config.timeout : 5000;
        this.config.maxMessages = !!this.config.maxMessages ? this.config.maxMessages : 5;
    };
    Object.defineProperty(AlertService.prototype, "messages", {
        get: /**
         * @return {?}
         */
        function () {
            return this.state.asObservable();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} msg
     * @return {?}
     */
    AlertService.prototype.info = /**
     * @param {?} msg
     * @return {?}
     */
    function (msg) {
        this.addAlert({ content: msg, type: 'info' });
    };
    /**
     * @param {?} msg
     * @return {?}
     */
    AlertService.prototype.danger = /**
     * @param {?} msg
     * @return {?}
     */
    function (msg) {
        this.addAlert({ content: msg, type: 'danger' });
    };
    /**
     * @param {?} msg
     * @return {?}
     */
    AlertService.prototype.success = /**
     * @param {?} msg
     * @return {?}
     */
    function (msg) {
        this.addAlert({ content: msg, type: 'success' });
    };
    /**
     * @param {?} msg
     * @return {?}
     */
    AlertService.prototype.warning = /**
     * @param {?} msg
     * @return {?}
     */
    function (msg) {
        this.addAlert({ content: msg, type: 'warning' });
    };
    /**
     * @param {?} alert
     * @return {?}
     */
    AlertService.prototype.close = /**
     * @param {?} alert
     * @return {?}
     */
    function (alert) {
        this.dispatcher.next({ fn: AlertReducer.remove, alert: alert, config: this.config });
    };
    /**
     * @private
     * @param {?} alert
     * @return {?}
     */
    AlertService.prototype.addAlert = /**
     * @private
     * @param {?} alert
     * @return {?}
     */
    function (alert) {
        var _this = this;
        this.dispatcher.next({ fn: AlertReducer.add, alert: alert, config: this.config });
        if (this.config.timeout > 0) {
            timer(this.config.timeout)
                .pipe(take(1))
                .subscribe((/**
             * @return {?}
             */
            function () {
                _this.dispatcher.next({ fn: AlertReducer.remove, alert: alert, config: _this.config });
            }));
        }
    };
    AlertService.decorators = [
        { type: Injectable }
    ];
    /** @nocollapse */
    AlertService.ctorParameters = function () { return [
        { type: undefined, decorators: [{ type: Inject, args: [ALERT_CONFIG,] }] }
    ]; };
    return AlertService;
}());
export { AlertService };
if (false) {
    /**
     * @type {?}
     * @private
     */
    AlertService.prototype.dispatcher;
    /**
     * @type {?}
     * @private
     */
    AlertService.prototype.state;
    /**
     * @type {?}
     * @private
     */
    AlertService.prototype.config;
}
