// ngx-alerts/esm2015/lib/alert.module.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { AlertComponent } from './component/alert.component';
import { AlertService } from './service/alert.service';
import { CommonModule } from '@angular/common';
import { ALERT_CONFIG } from './alert.config';
export class AlertModule {
    /**
     * @param {?=} config
     * @return {?}
     */
    static forRoot(config) {
        return {
            ngModule: AlertModule,
            providers: [
                AlertService,
                { provide: ALERT_CONFIG, useValue: config }
            ]
        };
    }
}
AlertModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    AlertComponent
                ],
                imports: [
                    CommonModule
                ],
                exports: [
                    AlertComponent
                ]
            },] }
];
