// ngx-spinners/esm2015/lib/components/bar-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { calculateRgba } from '../helpers/helpers';
export class BarLoaderComponent {
    constructor() {
        this.color = '#36d7b7';
        this.width = 100;
        this.widthUnit = 'px';
        this.height = 4;
        this.heightUnit = 'px';
    }
    /**
     * @return {?}
     */
    createBarWrapperStyle() {
        return {
            'position': 'relative',
            'width': `${`${this.width}${this.widthUnit}`}`,
            'height': `${`${this.height}${this.heightUnit}`}`,
            'overflow': 'hidden',
            'background-color': `${calculateRgba(this.color, 0.2)}`,
            'background-clip': 'padding-box',
        };
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createBarStyle(index) {
        return {
            'display': 'block',
            'position': 'absolute',
            'height': `${`${this.height}${this.heightUnit}`}`,
            'overflow': 'hidden',
            'background-color': `${this.color}`,
            'background-clip': 'padding-box',
            'border-radius': '2px',
            'will-change': 'left, right',
            'animation-fill-mode': 'forwards',
            'animation': `${index === 1 ? 'loading-long' : 'loading-short'} 2.1s ${index === 2 ? '1.15s' : ''}
                    ${index === 1 ? 'cubic-bezier(0.65, 0.815, 0.735, 0.395)' : 'cubic-bezier(0.165, 0.84, 0.44, 1)'} infinite`
        };
    }
}
BarLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-bar-loader',
                template: "<div class=\"loading\" [ngStyle]=\"createBarWrapperStyle()\">\r\n  <div [ngStyle]=\"createBarStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBarStyle(2)\"></div>\r\n</div>\r\n",
                styles: ["@-webkit-keyframes loading-long{0%{left:-35%;right:100%}100%,60%{left:100%;right:-90%}}@keyframes loading-long{0%{left:-35%;right:100%}100%,60%{left:100%;right:-90%}}@-webkit-keyframes loading-short{0%{left:-200%;right:100%}100%,60%{left:107%;right:-8%}}@keyframes loading-short{0%{left:-200%;right:100%}100%,60%{left:107%;right:-8%}}"]
            }] }
];
/** @nocollapse */
BarLoaderComponent.ctorParameters = () => [];
BarLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    width: [{ type: Input }],
    widthUnit: [{ type: Input }],
    height: [{ type: Input }],
    heightUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    BarLoaderComponent.prototype.color;
    /** @type {?} */
    BarLoaderComponent.prototype.width;
    /** @type {?} */
    BarLoaderComponent.prototype.widthUnit;
    /** @type {?} */
    BarLoaderComponent.prototype.height;
    /** @type {?} */
    BarLoaderComponent.prototype.heightUnit;
}
