// ngx-spinners/esm2015/lib/components/bounce-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class BounceLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 60;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    createBounceWrapperStyle() {
        return {
            'position': 'relative',
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`
        };
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createBounceStyle(index) {
        return {
            'position': 'absolute',
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'background-color': `${this.color}`,
            'border-radius': '100%',
            'opacity': '0.6',
            'top': '0',
            'left': '0',
            'animation-fill-mode': 'both',
            'animation': `loading-bounce 2.1s ${index === 1 ? '1s' : '0s'} infinite ease-in-out`
        };
    }
}
BounceLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-bounce-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createBounceWrapperStyle()\">\r\n  <div [ngStyle]=\"createBounceStyle(1)\"></div>\r\n  <div [ngStyle]=\"createBounceStyle(2)\"></div>\r\n</div>\r\n",
                styles: ["@-webkit-keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}@keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}"]
            }] }
];
/** @nocollapse */
BounceLoaderComponent.ctorParameters = () => [];
BounceLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    BounceLoaderComponent.prototype.color;
    /** @type {?} */
    BounceLoaderComponent.prototype.size;
    /** @type {?} */
    BounceLoaderComponent.prototype.sizeUnit;
}
