// ngx-spinners/esm2015/lib/components/circle-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class CircleLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 50;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    createCircleWrapperStyle() {
        return {
            'position': 'relative',
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`
        };
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createCircleStyle(index) {
        return {
            'position': 'absolute',
            'height': `${`${this.size * (1 - (index / 10))}${this.sizeUnit}`}`,
            'width': `${`${this.size * (1 - (index / 10))}${this.sizeUnit}`}`,
            'border': `1px solid ${this.color}`,
            'border-radius': '100%',
            'transition': '2s',
            'border-bottom': 'none',
            'border-right': 'none',
            'top': `${(index * 0.7 * 2.5)}%`,
            'left': `${(index * 0.35 * 2.5)}%`,
            'animation-fill-mode': '',
            'animation': `loading-circle 1s ${index * 0.2}s infinite linear`
        };
    }
}
CircleLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-circle-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createCircleWrapperStyle()\">\r\n  <div [ngStyle]=\"createCircleStyle(0)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(1)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(2)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(3)\"></div>\r\n  <div [ngStyle]=\"createCircleStyle(4)\"></div>\r\n</div>\r\n",
                styles: ["@-webkit-keyframes loading-circle{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}@keyframes loading-circle{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}"]
            }] }
];
/** @nocollapse */
CircleLoaderComponent.ctorParameters = () => [];
CircleLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    CircleLoaderComponent.prototype.color;
    /** @type {?} */
    CircleLoaderComponent.prototype.size;
    /** @type {?} */
    CircleLoaderComponent.prototype.sizeUnit;
}
