// ngx-spinners/esm2015/lib/components/clip-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class ClipLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 35;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    createClipStyle() {
        return {
            'background': 'transparent !important',
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'border-radius': '100%',
            'border': '2px solid',
            'border-color': `${this.color}`,
            'border-bottom-color': 'transparent',
            'display': 'inline-block',
            'animation': ' loading-clip 0.75s 0s infinite linear',
            'animation-fill-mode': 'both'
        };
    }
}
ClipLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-clip-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createClipStyle()\"></div>\r\n",
                styles: ["@-webkit-keyframes loading-clip{0%{transform:rotate(0) scale(1)}50%{transform:rotate(180deg) scale(.8)}100%{transform:rotate(360deg) scale(1)}}@keyframes loading-clip{0%{transform:rotate(0) scale(1)}50%{transform:rotate(180deg) scale(.8)}100%{transform:rotate(360deg) scale(1)}}"]
            }] }
];
/** @nocollapse */
ClipLoaderComponent.ctorParameters = () => [];
ClipLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    ClipLoaderComponent.prototype.color;
    /** @type {?} */
    ClipLoaderComponent.prototype.size;
    /** @type {?} */
    ClipLoaderComponent.prototype.sizeUnit;
}
