// ngx-spinners/esm2015/lib/components/dot-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class DotLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 60;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    createDotWrapperStyle() {
        return {
            'position': 'relative',
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'animation-fill-mode': 'forwards',
            'animation': 'loading-rotate 2s 0s infinite linear'
        };
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createDotStyle(index) {
        return {
            'position': 'absolute',
            'top': `${index % 2 ? '0' : 'auto'}`,
            'bottom': `${index % 2 ? 'auto' : '0'}`,
            'height': `${`${this.size / 2}${this.sizeUnit}`}`,
            'width': `${`${this.size / 2}${this.sizeUnit}`}`,
            'background-color': `${this.color}`,
            'border-radius': '100%',
            'animation-fill-mode': 'forwards',
            'animation': `loading-bounce 2s ${index === 2 ? '-1s' : '0s'} infinite linear`
        };
    }
}
DotLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-dot-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createDotWrapperStyle()\">\r\n  <div [ngStyle]=\"createDotStyle(1)\"></div>\r\n  <div [ngStyle]=\"createDotStyle(2)\"></div>\r\n</div>\r\n",
                styles: ["@-webkit-keyframes loading-rotate{100%{transform:rotate(360deg)}}@keyframes loading-rotate{100%{transform:rotate(360deg)}}@-webkit-keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}@keyframes loading-bounce{0%,100%{transform:scale(0)}50%{transform:scale(1)}}"]
            }] }
];
/** @nocollapse */
DotLoaderComponent.ctorParameters = () => [];
DotLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    DotLoaderComponent.prototype.color;
    /** @type {?} */
    DotLoaderComponent.prototype.size;
    /** @type {?} */
    DotLoaderComponent.prototype.sizeUnit;
}
