// ngx-spinners/esm2015/lib/components/moon-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class MoonLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 60;
        this.sizeUnit = 'px';
    }
    /**
     * @return {?}
     */
    createMoonWrapperStyle() {
        return {
            'position': 'relative',
            'width': `${`${this.size + ((this.size / 7) * 2)}${this.sizeUnit}`}`,
            'height': `${`${this.size + ((this.size / 7) * 2)}${this.sizeUnit}`}`,
            'animation': 'loading-moon 0.6s 0s infinite linear',
            'animation-fill-mode': 'forwards'
        };
    }
    /**
     * @return {?}
     */
    createMoonBallStyle() {
        return Object.assign({}, this.createBallStyle(this.size / 7), { 'background-color': `${this.color}`, 'opacity': '0.8', 'position': 'absolute', 'top': `${`${(this.size / 2) - (this.size / 14)}${this.sizeUnit}`}`, 'animation': 'loading-moon 0.6s 0s infinite linear', 'animation-fill-mode': 'forwards' });
    }
    /**
     * @return {?}
     */
    createMoonCircleStyle() {
        return Object.assign({}, this.createBallStyle(this.size), { 'border': `${this.size / 7}px solid ${this.color}`, 'opacity': '0.1' });
    }
    /**
     * @private
     * @param {?} size
     * @return {?}
     */
    createBallStyle(size) {
        return {
            'width': `${`${size}${this.sizeUnit}`}`,
            'height': `${`${size}${this.sizeUnit}`}`,
            'border-radius': '100%',
            'box-sizing': 'content-box'
        };
    }
}
MoonLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-moon-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createMoonWrapperStyle()\">\r\n  <div [ngStyle]=\"createMoonBallStyle()\"></div>\r\n  <div [ngStyle]=\"createMoonCircleStyle()\"></div>\r\n</div>\r\n",
                styles: ["@-webkit-keyframes loading-moon{100%{transform:rotate(360deg)}}@keyframes loading-moon{100%{transform:rotate(360deg)}}"]
            }] }
];
/** @nocollapse */
MoonLoaderComponent.ctorParameters = () => [];
MoonLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    MoonLoaderComponent.prototype.color;
    /** @type {?} */
    MoonLoaderComponent.prototype.size;
    /** @type {?} */
    MoonLoaderComponent.prototype.sizeUnit;
}
