// ngx-spinners/esm2015/lib/components/pacman-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, Input } from '@angular/core';
export class PacmanLoaderComponent {
    /**
     * @param {?} elementRef
     */
    constructor(elementRef) {
        this.elementRef = elementRef;
        this.color = 'rgb(54, 215, 183)';
        this.size = 25;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @param {?} changes
     * @return {?}
     */
    ngOnChanges(changes) {
        const { size, sizeUnit } = changes;
        if (size && !isNaN(size.currentValue) && sizeUnit && typeof sizeUnit.currentValue === 'string') {
            this.elementRef.nativeElement.style.setProperty('--tx', `${`${-4 * size.currentValue}${sizeUnit.currentValue}`}`);
            this.elementRef.nativeElement.style.setProperty('--ty', `${`${(-1 * size.currentValue) / 4}${sizeUnit.currentValue}`}`);
        }
    }
    /**
     * @return {?}
     */
    createPacmanWrapperStyle() {
        return {
            'position': 'relative',
            'font-size': '0',
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'width': `${`${this.size}${this.sizeUnit}`}`
        };
    }
    /**
     * @return {?}
     */
    createPacStyle() {
        return this.createPacManStyle(0);
    }
    /**
     * @return {?}
     */
    createManStyle() {
        return this.createPacManStyle(1);
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createBallStyle(index) {
        return {
            'width': `${`${this.size / 3}${this.sizeUnit}`}`,
            'height': `${`${this.size / 3}${this.sizeUnit}`}`,
            'background-color': `${this.color}`,
            'margin': `${`${this.margin}${this.marginUnit}`}`,
            'border-radius': '100%',
            'transform': `translate(0, ${`${-this.size / 4}${this.sizeUnit}`})`,
            'position': 'absolute',
            'top': `${this.size}px`,
            'left': `${`${this.size * 4}${this.sizeUnit}`}`,
            'animation': `loading-ball 1s ${index * 0.25}s infinite linear`,
            'animation-fill-mode': 'both'
        };
    }
    /**
     * @private
     * @param {?} index
     * @return {?}
     */
    createPacManStyle(index) {
        /** @type {?} */
        const s1 = this.s1();
        /** @type {?} */
        const s2 = this.s2();
        return {
            'width': '0',
            'height': '0',
            'border-right': `${s1}`,
            'border-top': `${index === 0 ? s1 : s2}`,
            'border-left': `${s2}`,
            'border-bottom': `${index === 0 ? s2 : s1}`,
            'border-radius': `${`${this.size}${this.sizeUnit}`}`,
            'position': 'absolute',
            'animation': `${'loading-pacman-' + index} 0.8s infinite ease-in-out`,
            'animation-fill-mode': 'both'
        };
    }
    /**
     * @private
     * @return {?}
     */
    s1() {
        return `${this.size}${this.sizeUnit} solid transparent`;
    }
    /**
     * @private
     * @return {?}
     */
    s2() {
        return `${this.size}${this.sizeUnit} solid ${this.color}`;
    }
}
PacmanLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-pacman-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createPacmanWrapperStyle()\">\r\n  <div [ngStyle]=\"createPacStyle()\"></div>\r\n  <div [ngStyle]=\"createManStyle()\"></div>\r\n  <div [ngStyle]=\"createBallStyle(2)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(3)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(4)\"></div>\r\n  <div [ngStyle]=\"createBallStyle(5)\"></div>\r\n</div>\r\n",
                styles: [":host{--tx:calc(-4 * 25px);--ty:calc((-1 * 25px) / 4)}@-webkit-keyframes loading-ball{75%{opacity:.7}100%{transform:translate(var(--tx),var(--ty))}}@keyframes loading-ball{75%{opacity:.7}100%{transform:translate(var(--tx),var(--ty))}}@-webkit-keyframes loading-pacman-0{0%{transform:rotate(0)}50%{transform:rotate(-44deg)}}@keyframes loading-pacman-0{0%{transform:rotate(0)}50%{transform:rotate(-44deg)}}@-webkit-keyframes loading-pacman-1{0%{transform:rotate(0)}50%{transform:rotate(44deg)}}@keyframes loading-pacman-1{0%{transform:rotate(0)}50%{transform:rotate(44deg)}}"]
            }] }
];
/** @nocollapse */
PacmanLoaderComponent.ctorParameters = () => [
    { type: ElementRef }
];
PacmanLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }],
    margin: [{ type: Input }],
    marginUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PacmanLoaderComponent.prototype.color;
    /** @type {?} */
    PacmanLoaderComponent.prototype.size;
    /** @type {?} */
    PacmanLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    PacmanLoaderComponent.prototype.margin;
    /** @type {?} */
    PacmanLoaderComponent.prototype.marginUnit;
    /**
     * @type {?}
     * @private
     */
    PacmanLoaderComponent.prototype.elementRef;
}
