// ngx-spinners/esm2015/lib/components/pulse-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class PulseLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createPulseStyle(index) {
        return {
            'display': 'inline-block',
            'background-color': `${this.color}`,
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'margin': `${`${this.margin}${this.marginUnit}`}`,
            'border-radius': '100%',
            'animation': `loading-pulse 0.75s ${index * 0.12}s infinite cubic-bezier(.2,.68,.18,1.08)`,
            'animation-fill-mode': 'both'
        };
    }
}
PulseLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-pulse-loader',
                template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createPulseStyle(1)\"></div>\r\n  <div [ngStyle]=\"createPulseStyle(2)\"></div>\r\n  <div [ngStyle]=\"createPulseStyle(3)\"></div>\r\n</div>\r\n",
                styles: ["@-webkit-keyframes loading-pulse{0%,80%{transform:scale(1);opacity:1}45%{transform:scale(.1);opacity:.7}}@keyframes loading-pulse{0%,80%{transform:scale(1);opacity:1}45%{transform:scale(.1);opacity:.7}}"]
            }] }
];
/** @nocollapse */
PulseLoaderComponent.ctorParameters = () => [];
PulseLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }],
    margin: [{ type: Input }],
    marginUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PulseLoaderComponent.prototype.color;
    /** @type {?} */
    PulseLoaderComponent.prototype.size;
    /** @type {?} */
    PulseLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    PulseLoaderComponent.prototype.margin;
    /** @type {?} */
    PulseLoaderComponent.prototype.marginUnit;
}
