// ngx-spinners/esm2015/lib/components/rise-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
export class RiseLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @param {?} index
     * @return {?}
     */
    createRiseStyle(index) {
        return {
            'display': 'inline-block',
            'border-radius': '100%',
            'background-color': `${this.color}`,
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'margin': `${`${this.margin}${this.marginUnit}`}`,
            'animation': `${index % 2 === 0 ? 'loading-rise-even' : 'loading-rise-odd'} 1s 0s infinite cubic-bezier(.15,.46,.9,.6)`,
            'animation-fill-mode': 'both'
        };
    }
}
RiseLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-rise-loader',
                template: "<div class=\"loader\">\r\n  <div [ngStyle]=\"createRiseStyle(1)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(2)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(3)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(4)\"></div>\r\n  <div [ngStyle]=\"createRiseStyle(5)\"></div>\r\n</div>\r\n",
                encapsulation: ViewEncapsulation.ShadowDom,
                styles: [":host{--riseAmount:30px}@-webkit-keyframes loading-rise-even{0%{transform:scale(1.1)}50%{transform:scale(.4)}75%{transform:translateY(var(--riseAmount))}100%{transform:translateY(0) scale(1)}}@keyframes loading-rise-even{0%{transform:scale(1.1)}50%{transform:scale(.4)}75%{transform:translateY(var(--riseAmount))}100%{transform:translateY(0) scale(1)}}@-webkit-keyframes loading-rise-odd{0%{transform:scale(.4)}50%{transform:scale(1.1)}75%{transform:translateY(calc(-1 * var(--riseAmount)))}100%{transform:translateY(0) scale(.75)}}@keyframes loading-rise-odd{0%{transform:scale(.4)}50%{transform:scale(1.1)}75%{transform:translateY(calc(-1 * var(--riseAmount)))}100%{transform:translateY(0) scale(.75)}}"]
            }] }
];
/** @nocollapse */
RiseLoaderComponent.ctorParameters = () => [];
RiseLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }],
    margin: [{ type: Input }],
    marginUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    RiseLoaderComponent.prototype.color;
    /** @type {?} */
    RiseLoaderComponent.prototype.size;
    /** @type {?} */
    RiseLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    RiseLoaderComponent.prototype.margin;
    /** @type {?} */
    RiseLoaderComponent.prototype.marginUnit;
}
