// ngx-spinners/esm2015/lib/components/rotate-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input, ViewEncapsulation } from '@angular/core';
export class RotateLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @return {?}
     */
    createRotateWrapperStyle() {
        return Object.assign({ 'display': 'inline-block' }, this.createBallStyle(), { 'position': 'relative', 'animation-fill-mode': 'both', 'animation': 'loading-rotate 1s 0s infinite cubic-bezier(.7,-.13,.22,.86)' });
    }
    /**
     * @return {?}
     */
    createRotateLongStyle() {
        return Object.assign({}, this.createBallStyle(), this.createPositionStyle(1));
    }
    /**
     * @return {?}
     */
    createRotateShortStyle() {
        return Object.assign({}, this.createBallStyle(), this.createPositionStyle(2));
    }
    /**
     * @private
     * @return {?}
     */
    createBallStyle() {
        return {
            'background-color': `${this.color}`,
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'margin': `${`${this.margin}${this.marginUnit}`}`,
            'border-radius': '100%'
        };
    }
    /**
     * @private
     * @param {?} index
     * @return {?}
     */
    createPositionStyle(index) {
        return {
            'opacity': '0.8',
            'position': 'absolute',
            'top': '0',
            'left': `${index % 2 ? -28 : 25}px`
        };
    }
}
RotateLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-rotate-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createRotateWrapperStyle()\">\r\n  <div [ngStyle]=\"createRotateLongStyle()\"></div>\r\n  <div [ngStyle]=\"createRotateShortStyle()\"></div>\r\n</div>\r\n",
                encapsulation: ViewEncapsulation.ShadowDom,
                styles: ["@-webkit-keyframes loading-rotate{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}@keyframes loading-rotate{0%{transform:rotate(0)}50%{transform:rotate(180deg)}100%{transform:rotate(360deg)}}"]
            }] }
];
/** @nocollapse */
RotateLoaderComponent.ctorParameters = () => [];
RotateLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }],
    margin: [{ type: Input }],
    marginUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    RotateLoaderComponent.prototype.color;
    /** @type {?} */
    RotateLoaderComponent.prototype.size;
    /** @type {?} */
    RotateLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    RotateLoaderComponent.prototype.margin;
    /** @type {?} */
    RotateLoaderComponent.prototype.marginUnit;
}
