// ngx-spinners/esm2015/lib/components/square-loader.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
export class SquareLoaderComponent {
    constructor() {
        this.color = 'rgb(54, 215, 183)';
        this.size = 15;
        this.sizeUnit = 'px';
        this.margin = 2;
        this.marginUnit = 'px';
    }
    /**
     * @return {?}
     */
    createBoxStyle() {
        return {
            'display': 'inline-block',
            'background-color': `${this.color}`,
            'width': `${`${this.size}${this.sizeUnit}`}`,
            'height': `${`${this.size}${this.sizeUnit}`}`,
            'animation': 'loading-square 3s 0s infinite cubic-bezier(.09,.57,.49,.9)',
            'animation-fill-mode': 'both'
        };
    }
}
SquareLoaderComponent.decorators = [
    { type: Component, args: [{
                selector: 'ngx-square-loader',
                template: "<div class=\"loader\" [ngStyle]=\"createBoxStyle()\"></div>\r\n",
                styles: ["@-webkit-keyframes loading-square{25%{transform:rotateX(180deg) rotateY(0)}50%{transform:rotateX(180deg) rotateY(180deg)}75%{transform:rotateX(0) rotateY(180deg)}100%{transform:rotateX(0) rotateY(0)}}@keyframes loading-square{25%{transform:rotateX(180deg) rotateY(0)}50%{transform:rotateX(180deg) rotateY(180deg)}75%{transform:rotateX(0) rotateY(180deg)}100%{transform:rotateX(0) rotateY(0)}}"]
            }] }
];
/** @nocollapse */
SquareLoaderComponent.ctorParameters = () => [];
SquareLoaderComponent.propDecorators = {
    color: [{ type: Input }],
    size: [{ type: Input }],
    sizeUnit: [{ type: Input }],
    margin: [{ type: Input }],
    marginUnit: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    SquareLoaderComponent.prototype.color;
    /** @type {?} */
    SquareLoaderComponent.prototype.size;
    /** @type {?} */
    SquareLoaderComponent.prototype.sizeUnit;
    /** @type {?} */
    SquareLoaderComponent.prototype.margin;
    /** @type {?} */
    SquareLoaderComponent.prototype.marginUnit;
}
